\chapter{Software Tools for Image and Sequence Analysis}

\todo[inline]{Add one page general background to broad needs and
  motivations of software.  Two sections on: Why software development
  is essential for cell biology (1 page), and Why FOSS is important for
  researchers (half page)}

\section{GNU Octave}

The GNU Octave programming language was heavily used for the analysis
of microscope images throughout this work.  Several features of the
language and its supportive community make it an attractive choice
for qualitative microscopy.

\todo[inline]{Add 2 sentences with a general description of Octave
  capabilities}

Biological microscope images have a varying number of dimensions.
In addition to the 2 dimensional plane of a standard image, microscope
outputs typically include any combination of: $z$ dimension along
the optical axis for a volume image; time dimension for time-lapse
experiments; and wavelength for multi-channel experiments.  Recent microscopy
techniques generate images with an even higher number of dimensions
such as angle, phase, and lifetime.  This varying dimensionality adds
complexity and makes it more challenging to write generalized subroutines for
microscope image analysis.

GNU Octave is an array programming language primarily intended for
numerical computations \citep{octave}.  Array programming languages
are high level programming languages where the primary data types and
operators \todo{explain data types and operators}
are generalised to multidimensional arrays.  This provides
an abstraction layer that is useful when writing code for an arbitrary
number of dimensions.

%% Doesn't seem to be any scientific study about REPL as an advantage
%% for prototyping.  Similarly, none about being better for
%% exploratory data analysis but I can't imagine anyone disagreeing
%% with it.
Octave also has a read–eval–print loop (REPL), or interactive top
level, like Lisp machines, IPython, and Unix shells, which reduces the
feedback time and provides an efficient
environment for exploratory data analysis.

%% No citation for this, only the anecdotal evidence of being the
%% project leader of octave Forge for 6 years.
The project has a large and active community
mainly composed of
scientists and engineers as a large support group of
specialists in numerical computations

GNU Octave and Octave Forge packages are free software which allows
the study, modification, and distribution of modified source.
We made extensive
use of this feature and have contributed to improving Octave and its
packages for the needs of quantitative microscopy.

In addition, Octave has
packages that focus on specific applications such as control
systems \citep{octave-control}, time-frequency analysis
\citep{octave-ltfat}, or level sets \citep{octave-level-set}.  The
Octave Forge project hosts
Octave packages and provides a collaborative environment for their
development.  The Octave Forge image package
provides a number of functions for image processing such as geometric
transformations, mathematical morphology, image registration, and
noise reduction.

%% These were actually lsm files but those are actually TIFFs.
While Octave is well suited to image
processing, we also identified several problems always related with
the large image size or number of dimensions.  Our microscopy images
were several megabytes in size.  For example, the single cell
FRAP experiments for histone dynamics generating TIFF files of 449
These images had a field of view of 300
by 300 pixels and 2500 time frames
on acquisition by an 8-bit camera.
Such files sizes are typical in the field of microscopy.

\subsection{Reading and Writing of Image Files}

%% Number of 88 major image formats taken from GraphicsMagick
%% documentation on February 2017.
Octave uses the GraphicsMagick C++ library for the reading and writing
of image files which provides a common API \todo{define API}
to a varied collection of
image format specific libraries covering almost
90 major image formats.  In Octave, this functionality is
provided via the functions \command{imfinfo}, \command{imread}, and
\command{imwrite}.

These three functions were rewritten with the aim of achieving
reduced memory usage, increased performance, improved
interface for multidimensional images, and new image types.
New options were added to read and write a series of image planes
with in a single function call, to read specific regions of interest in
individual planes, to read and write images
with floating point precision, and to append additional planes to existing image
files.

As part of this rewrite, existing features were improved such as
support for transparency and
indexed images, support for the CMYK colour model, EXIF
and GPS metadata, reading and writing of animations in GIF images, and
control over the image compression type.

To support this development, a system of configurable hooks \todo{explain hooks}
to the image IO \todo{define IO}
functions were added based on individual file formats.
This enables plugin \todo{explain plugin} support for new image file formats
without addition of new format specific functions, and improved access
to the existing formats.  For example, it is now possible to read
microscope specific metadata from \command{imfinfo}.
This system is available via the function \command{imformats}.

Changes extended to other functions of Octave related to images
including overhaul of functions involved in conversion between color
models, grayscale images, and indexed images to support integer and
floating point data types, and multiple dimensions.

All changes were released with Octave version 3.8.0.

\begin{table}
  \captionIntro{New or improved functions in GNU Octave}{}
  \begin{multicols}{3}
    \begin{itemize}
      \foreach \function in {
        bitcmp,
        bzip2,
        cubehelix,
        fliplr,
        flipud,
        flip,
        frame2im,
        gallery,
        gray2ind,
        gzip,
        hsv2rgb,
        im2double,
        im2frame,
        imfinfo,
        imformats,
        imread,
        imwrite,
        ind2gray,
        ind2rgb,
        inputParser,
        ntsc2rgb,
        pkg,
        psi,
        rectint,
        rgb2hsv,
        rgb2ind,
        rgb2ntsc,
        rot90,
        validateattributes}
      { \item \command{\function} }
    \end{itemize}
  \end{multicols}
\end{table}

\subsubsection{Bio-Formats and Octave Java interface}

%% We kind of managed to read the pixel data.  This also required
%% knowing that LSM interleaves the actual pixel data with thumbnails
%% which need to be skipped.
While GraphicsMagick provides support to read many image formats, its
support for scientific microscope image formats is limited.
For example, Zeiss confocal microscopes save
images in the LSM file format which is a proprietary extension of TIFF.
Although this reads pixel data,
the file metadata such as pixel size, time interval, or region of
bleaching event cannot be retrieved.

Bio-Formats is a free software library for reading and writing image
data with a strong focus on microscopy image file formats
\citep{bioformats}.  It is written in the Java programming language and
used by other programs in the field of microscope image analysis such
as CellProfiler \citep{cellprofiler}, ImageJ \cite{imagej2}, and OMERO
\citep{omero}.

%% Before version 3.8.0, there was a separate Octave Forge package
%% that added the java interface.  Actually, Octave version 3.8.0
%% pretty much only merged the java package into it.
\todo{explain interface in 1 sentence}
Octave has a native Java interface that should enable integration with
Bio-Formats.
We identified a series of problems in the Octave Java interface that
could be solved either by improving Octave or Bio-Formats.

In Octave, we rewrote the wrapping and unwrapping of Java objects \todo{explain}
into Octave data types.  This enabled Octave to automatically
convert Java objects into native Octave data types.
Support for arrays of Java arrays was not enabled
since we did not require it and Octave still lacks
support for it.

In Bio-Formats, we modified the code to use a Java interface that is
both Octave and Matlab compatible.
This change considerably  simplified the packaging of
Bio-Formats for Matlab and Octave so that they are now
effectively the same code.
Finally, we configured the Bio-Formats build system \todo{explain}
to routinely prepare an Octave package as part of its releases.

All changes were released with Octave since version 4.0.0 and
Bio-Formats since version 5.1.2

\begin{table}
  \captionIntro{New or improved functions in the Octave Forge image
    package}{}
  \begin{multicols}{3}
    \begin{itemize}
      \foreach \function in {
        bestblk,
        bwareafilt,
        bwareaopen,
        bwconncomp,
        bwdist,
        bwlabel,
        bwlabeln,
        bwmorph,
        bwperim,
        bwpropfilt,
        checkerboard,
        col2im,
        colfilt,
        conndef,
        edgetaper,
        fftconv2,
        fftconvn,
        grayslice,
        graythresh,
        im2col,
        imabsdiff,
        imadjust,
        imattributes,
        imbothat,
        imclearborder,
        imclose,
        imcomplement,
        imcrop,
        imdilate,
        imerode,
        imfill,
        imgetfile,
        imhist,
        imlincomb,
        immse,
        imopen,
        impixel,
        impyramid,
        imquantize,
        imreconstruct,
        imregionalmax,
        imregionalmin,
        imresize,
        imrotate,
        imtophat,
        intlut,
        iptcheckconn,
        label2rgb,
        labelmatrix,
        mat2gray,
        mmgradm,
        montage,
        nlfilter,
        normxcorr2,
        ordfiltn,
        otf2psf,
        padarray,
        psf2otf,
        psnr,
        regionprops,
        rgb2ycbcr,
        strel,
        stretchlim,
        subimage,
        tiff\_tag\_read,
        watershed,
        wavelength2rgb,
        ycbcr2rgb}
      { \item \command{\function} }
  \end{itemize}
  \end{multicols}
\end{table}

\subsubsection{Octave Forge image package}

\subsubsection{Octave Forge zenity package}

\subsubsection{Other Octave Forge packages}

\section{BioPerl}

The BioPerl project is an international association of developers of
free Perl software for bioinformatics, genomics, and life science
\citep{bioperl}.  It has created the BioPerl distribution of Perl
modules which has almost 800 modules for management and manipulation
of biological data, from protein and genomic sequences, gene
ontologies, and bibliography references, to programmatic access of
databases such as GenBank and SwissProt, and bioinformatics tools such
as ClustalW and Blast+.

%% $ find bioperl-live/Bio/ -name '*.pm' -type f | wc -l
%% 791
%%
%% This number no longer includes the modules already moved out of
%% BioPerl and into their own distributions.

\subsection{Dist::Zilla and Pod::Weaver}

%% Technically, the first one to be moved out of BioPerl was
%% Bio-Graphics and that was in 2009.  However, that was an
%% independent event, not part of a concerted effort of reorganising
%% BioPerl.  That only started in 2011 (see Changes file in
%% bioperl-live).
The large number of modules in BioPerl became a maintenance problem
and in 2011, a new project was started to split BioPerl into smaller
more manageable distributions such as Bio-Biblio, Bio-FeatureIO, and
Bio-Coordinate.  With the aim of of preventing duplication of
maintainers code, reducing that existing code, and making new release
a further easier process, Dist::Zilla and Pod::Weaver were adopted for
the new distributions.

Dist::Zilla is a program to make it easier to write, package, manage,
and release free software, targeted at libraries written in the Perl
programming language and released to the Comprehensive Perl Archive
Network (CPAN).  It comes with a series of plugins to automate the
release process such as the addition of copyright notices, discovery
of dependencies, and uploading to CPAN.

Pod::Weaver is a program to create documents in Plain Old
Documentation (POD) format, a format mainly used to write
documentation inlined in Perl modules.  It has a Dist::Zilla plugin
that serves as bridge between the two, so that most boilerplate POD is
generated automatically as part of the release process.

As part of the restructure of the BioPerl distribution, we configured
the BioPerl Dist::Zilla PluginBundle, and created two new Pod::Weaver
section plugins, GenerateSection and Legal::Complicated.

GenerateSection creates a POD sections based on a templates.  It is
used in BioPerl to generate the support section of individual modules
with distribution specific details such as links to the online
repository.

Legal::Complicated creates a POD section for the copyright details
based on comments in individual modules.  While BioPerl is free
software, individual modules may be released under different free
software licenses, and each has their own author which often is not
the copyright holder.

Both new Pod::Weaver plugins and the BioPerl PluginBundle are available
on CPAN.

\subsection{Debian packaging}

Debian is a computer operating system, the set of low-level software
that manages a computer hardware and resources for computer programs,
composed entirely of free software and one of the earliest GNU/Linux
distributions.  Like all free operating systems today, Debian is
package based meaning that it is made of multiple components, the
packages.  For example, in Debian there are packages for the Linux
kernel, the Perl programming language, and Dist::Zilla.  Packages are
managed by a package management system which handles their
installation, configuration, and removal, tasks which otherwise would
have to be handled by the user.

While Debian had a package for the main BioPerl distribution, it did
not have one for Bio-EUtilities.  With the aim of making it easier for
others to reproduce our results, we packaged it for Debian.
Similarly, with the aim of making it easier for prospective BioPerl
developers, we packaged all the Dist::Zilla plugins required, as well
as all the module distributions required by those
\trefp{tab:software:debian-packages}.

\begin{table}
  \captionIntro{Perl module distributions packaged for Debian}{}
  \label{tab:software:debian-packages}
  \begin{itemize}
  \item Bio-EUtilities
  \item Config-MVP-Slicer
  \item Dist-Zilla-Config-Slicer
  \item Dist-Zilla-Plugin-AutoMetaResources
  \item Dist-Zilla-Plugin-MojibakeTests
  \item Dist-Zilla-Plugin-ReadmeFromPod
  \item Dist-Zilla-Plugin-Test-Compile
  \item Dist-Zilla-Role-PluginBundle-PluginRemover
  \item MooseX-Types-Email
  \item Pod-Weaver-Plugin-EnsureUniqueSections
  \item Pod-Weaver-Section-Contributors
  \item Pod-Weaver-Section-GenerateSection
  \item Pod-Weaver-Section-Legal-Complicated
  \item Test-Mojibake
  \end{itemize}
\end{table}

%% Debian Jessie has 21024 source packages.  The binary number of
%% packages would be much higher but while that number is what is
%% usually used when making a point for Debian's high number of
%% packages, it is misleading.  The point is how much of upstream
%% projects are packaged, and that is the number of source packages.

The choice of Debian was strategic.  Debian is a widely used GNU/Linux
distribution with more than 21000 packages, but it also has a large
number of derivative distributions.  These are new distributions that
inherit the base of their packages from Debian, and some, like Ubuntu
and Knoppix, are the parents of their own derivatives.  By packaging
for Debian, we effectively prepare packages for the whole family of
Debian based distributions.

\subsection{Bio-EUtilities}

For the work in \Cref{ch:histone-catalogue}, we required a tool to
automate the search of histone genes, and download all sequences
associated with them.  We used the Gene database for the searches and
created a new program using BioPerl's Bio-EUtilities.

Gene is a public database hosted at the National Center for
Biotechnology Information (NCBI), which maps known or predicted genes
to other entries in the NCBI Reference Sequence (RefSeq), namely to
the Genome, Nucleotide, and Protein databases, which provide genomic,
transcript, and proteins sequences respectively \citep{gene-database}.

Bio-EUtilities is part of the BioPerl project and provides a Perl
interface to NCBI's Entrez Programming Utilities (E-Utilities).
Entrez is a federated search engine for multiple databases of
biomedical data, one of which is the Gene database.  Entrez has an
interactive interface at \url{https://www.ncbi.nlm.nih.gov/} while
E-Utilities provides a programming interface for queries using a fixed
URL syntax.

The program created requires a query for the Entrez Gene database and
then downloads all genomic, transcript, and protein sequences, as well
as a CSV file with chromosome coordinates, names, and identifiers.  It
has several options such as download of flanking sequences, different
output format, choice of genome assembly, and skipping of non coding
genes.  It is provided with an extensive manual covering all options
and examples \Crefp{app:pod-doc}.

The program bp\_genbank\_ref\_extractor was released with
Bio-EUtilities version 1.73.

\section{Build systems for scientific publications}
\subsection{SCons Perl tool}

One of the most commons and widespread build systems is GNU Make.
This was the basis for

However, for this work we used the SCons (from Software Construction)
build system \citep{scons}.  SCons was designed from the start to be a
replacement of Make, and it still resembles Make in concept, but has
the advantage of being configured by Python scripts, a modern
programming language often praised for its readability.

%% Other people may mention other advantages but I disagree:
%% 1.SCons has builtin configure and dependency analysis.  Well, that
%%   is bullshit.  First, it doesn't work properly.  Configure support
%%   was a second thought and is very much incomplete.  Second,
%%   scanner is really slow.  And while Make itself really does not
%%   have them, it is meant to be used as part of Autotools where
%%   those jobs are done by autoconf and automake.
%% 2. SCons uses a liberal licence.  Well, I prefer copyleft and
%%    anyway that's the build system.
%%
%% One advantage that I could agree is default to md5sums instead of
%% timestamps but then I would have to explain what md5 and a checksum
%% is.  It also has builtin support for latex but that's really easy
%% to do in GNU Make.

For the work in \Cref{ch:histone-catalogue} we made extensive use of
the Perl programming language which is not supported by default in
SCons.  We then created a SCons perl tool which we made available at
\url{https://bitbucket.org/carandraug/scons-perl5} under a free
software licence.  The SCons perl5 tool adds automatic prerequesite
scanning, perl configuration options, and multiple functions for using
Perl scripts.

%% automatic prerequesite scanning -> Scanner
%% configuration options -> Variables
%% multiple functions -> all the perl builders.
