%% We have what would be the manuscript for humans as a thesis chapter
%% and the supplementary figures as a chapter on the appendices.
%% However, we also want to show the results for mice.  For that, we
%% include what would be the catalogue as another appendix chapter.
%% However, there's a few things on the supplementary and manuscript
%% that we would also like to include on the mice chapter.  Those are
%% repeated here.

\newpage
\begin{center}
  \captionof{figure}[]{
    Sequence logo for the mice canonical H2A gene coding regions
    listed in table \tref{tab:histone-catalogue}.
    Initiator codon ATG and stop codon are omitted.
  }
  \label{fig:h2a-histone-gene-variation}
  \includegraphics[width=1.0\textwidth]{\FigsDir/seqlogo_H2A_cds.pdf}
\end{center}
\newpage
\begin{center}
  \captionof{figure}[]{
    Sequence logo for the mice canonical H2B gene coding regions
    listed in table \tref{tab:histone-catalogue}.
    Initiator codon ATG and stop codon are omitted.
  }
  \label{fig:h2b-histone-gene-variation}
  \includegraphics[width=1.0\textwidth]{\FigsDir/seqlogo_H2B_cds.pdf}
\end{center}
\newpage
\begin{center}
  \captionof{figure}[]{
    Sequence logo for the mice canonical H3 gene coding regions
    listed in table \tref{tab:histone-catalogue}.
    Initiator codon ATG and stop codon are omitted.
  }
  \label{fig:h3-histone-gene-variation}
  \includegraphics[width=1.0\textwidth]{\FigsDir/seqlogo_H3_cds.pdf}
\end{center}
\newpage
\begin{center}
  \captionof{figure}[]{
    Sequence logo for the mice canonical H4 gene coding regions
    listed in table \tref{tab:histone-catalogue}.
    Initiator codon ATG and stop codon are omitted.
  }
  \label{fig:h4-histone-gene-variation}
  \includegraphics[width=1.0\textwidth]{\FigsDir/seqlogo_H4_cds.pdf}
\end{center}

\newpage
\begin{center}
  \captionof{table}[]{
    Mean synonymous and non-synonymous distances between pairs of
    canonical core histone genes.  Mean over all pairwise comparisons
    of non-synonymous ($d_N$) and synonymous ($d_S$) nucleotide
    substitutions and mean of $d_N/d_S$ ratios for canonical core
    histone coding regions computed using \citet{GoldmanYang1994}
    model implemented in \texttt{codeml} from the PAML package
    \citep{PAML2007}.
  }
  \label{tab:histone-gene-differences}
  \begin{tabular}{l l l l}
    \toprule
    \null & \centercell{$d_N$} & \centercell{$d_S$} & \centercell{$d_N/d_S$} \\
    \midrule
    H2A & \MeanHTwoAdN  & \MeanHTwoAdS  & \MeanHTwoAdNdS \\
    H2B & \MeanHTwoBdN  & \MeanHTwoBdS  & \MeanHTwoBdNdS \\
    H3  & \MeanHThreedN & \MeanHThreedS & \MeanHThreedNdS \\
    H4  & \MeanHFourdN  & \MeanHFourdS  & \MeanHFourdNdS \\
    \bottomrule
  \end{tabular}
\end{center}

\newpage
\captionof{table}[]{%
  Codon usage frequency for each amino acid
  and canonical core histone type.
}
\label{tab:histone-gene-codonusage}
\input{\ResultsDir/table-codon_usage}

\newpage
\begin{figure}[h!]
  \centering
  \subtop[Stem-loop]{
    \includegraphics{\FigsDir/seqlogo_stem_loops.pdf}
    \label{fig:stem-loop-seqlogo}
  }
  \hfill
  \subtop[HDE]{
    \includegraphics{\FigsDir/seqlogo_HDEs.pdf}
    \label{fig:HDE-seqlogo}
  }
  \caption[]{%
    Sequence logos for
    \subcaptionref{fig:stem-loop-seqlogo} annotated stem-loops and
    \subcaptionref{fig:HDE-seqlogo}  Histone Downstream Elements (HDEs)
    identified by homology
    for all canonical core histone gene 3' untranslated regions (UTRs).
  }
\end{figure}

\newpage
\begin{center}
  \captionof{table}[]{
    Changes in mice canonical core histone gene catalogue
    annotated in NCBI RefSeq obtained on \SequencesDate{}
    compared to \citet{Marzluff02}.
  }
  \input{\ResultsDir/table-reference_comparison}
\end{center}
