\section{Results}

  \subsection{Porting of FRAP analysis}

    For estimation of binding constants from FRAP data, we used a
    previously written model for circle FRAP \citep{mueller2008evidence}.
    The original code was written for the \textsc{Matlab} which we
    ported to GNU Octave.

    While GNU Octave aims to be fully \textsc{Matlab} compatible, certain
    less used features were missing at the time the port was done.
    These could be split into two groups: the functions for nonlinear
    regression \texttt{nlinfit} and \texttt{nlparci} from the Statistics
    Toolbox; and functions for graphical user
    interaction such as \texttt{uigetfile} and \texttt{impoly}.

    Functions for nonlinear regression were required for the model
    fitting and were replaced by \texttt{leasqr}
    from Octave-Forge's optim package. This function performs the
    Levenberg--Marquardt nonlinear least squares algorithm which is the
    same as the documented method for \texttt{nlinfit}.
    Several sample images were analyzed
    by the Octave port and the fitted values were found to be equal to
    the values reported by the authors using the original code.

    Functions for graphical user interaction were mainly required
    for manual selection of Regions Of Interest (ROIs) and options.
    Selection of ROIs was replaced by methods capable of identifying
    them automatically. For setting of different options, a different
    approach was taken and a more typical stand-alone program for batch
    processing, capable of accepting options via command-line was prepared.

    \subsubsection{ROI selection}

      Three different ROIs are required for the used circle FRAP model:
      bleach spot, nucleus region, and background region. The bleach
      spot is required not only to measure the intensity recovery, but
      also to model the photobleach profile since it takes into account
      a non-uniform spatial distribution of the bleach spot. The nucleus
      region to model a finite sized nucleus and consider the fluorescence
      ``destroyed'' during the photobleaching. Finally, a small region
      outside the nucleus is used for background correction.

      Cells expressing GFP tagged histone proteins display a well defined
      nuclei. The high amount of histone protein
      and its specificity for nuclear chromatin, produces a high clear
      signal for the nuclei which contrasted on a dark background (\fref{fig:kill-frap:roi}).

      \begin{figure}
        \centering
        \subbottom[pre-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/roi-prebleach.png}
          \label{fig:kill-frap:prebleach}
        }
        \hfill
        \subbottom[post-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/roi-postbleach.png}
          \label{fig:kill-frap:postbleach}
        }
        \subbottom[pre-bleach $-$ post-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/roi-subtracted.png}
          \label{fig:kill-frap:subtracted}
        }
        \hfill
        \subbottom[Identified ROIs]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/roi-selected.png}
          \label{fig:kill-frap:selected}
        }
        \captionIntro{Automatic selection of ROIs for FRAP}
          {
            HeLa stable cell line expressing the H4~R45H mutant tagged with YFP,
            are imaged every \SI{30}{\ms} in a confocal microscope. A circular
            shape is used for photobleaching after 100~frames.
            \subcaptionref{fig:kill-frap:prebleach} averaging of 50~pre-bleach
            images removes most of the noise, allowing for a better refined
            ROI;
            \subcaptionref{fig:kill-frap:postbleach} average of
            5 post-bleach images;
            \subcaptionref{fig:kill-frap:subtracted} subtraction of the
            post-bleach to the pre-bleach image, gives a clear indication
            of the bleach spot, as well as faint signal for the nuclear
            region due to unintentional photobleaching;
            \subcaptionref{fig:kill-frap:prebleach} perimeter of the
            automatically identified ROIs superimposed on the pre-bleach
            image: cell nuclei, bleach spot, and background region.
          }
        \label{fig:kill-frap:roi}
      \end{figure}

      Subtraction of the post-bleach (\fref{fig:kill-frap:postbleach}) to
      the pre-bleach (\fref{fig:kill-frap:prebleach}) images displays
      a clear circular shape in a black background corresponding to
      the location of the bleach spot (\fref{fig:kill-frap:subtracted}).
      The centre for the bleach spot was
      identified by finding the maximum of the convolution matrix between
      the subtracted image and a disk kernel. For a reduced signal to noise
      ratio, several pre and post-bleach images were averaged before the
      operations. While the bleach spot is
      the most visible feature, there is also a faint signal on the nuclear
      region which is caused by unintentional photobleaching as result of
      the imaging.

      A rough border of the cell nuclei could be easily defined by
      Otsu's method for image threshold. Similar to the method used
      for identification of the bleach spot, several pre-bleach images
      were averaged before the threshold for a more refined cell border (\fref{fig:kill-frap:selected}).
      Since multiple nuclei often appeared in a single image,
      the coordinates for the bleach spot are used as reference to select
      the correct nucleus.

      The background region was identified by finding the minimum of
      the convolution matrix between the averaged pre-bleach images and
      a small square of black intensity values.

    \subsubsection{Batch processing}

      \begin{sidewaysfigure}
        \includegraphics[width=\textwidth]{kill-frap/frapinator.png}
        \captionIntro{Frapinator visual log files for batch processing}
          {
            Each FRAP experiment generates a log file with 6 different plots
            displaying the analysed values and the fitting to different models.
            In conjunction with the images in \fref{fig:kill-frap:roi} this
            provides a quick overview of the entire analysis process.
            The top left plot displays the raw intensity
            for the background, bleach spot, and nucleus intensity over the
            duration of the FRAP experiment. This is followed by the normalized
            intensity for the bleach spot which is actually used for the
            fitting. The top right displays the intensity
            profile for the bleach spot, and its fit to a radial profile
            model. The three bottom panels display the data fitted to three
            different models: pure diffusion which has no terms for binding
            constants; full model with a fixed diffusion rate; and full model
            with all the 3 terms.
          }
        \label{fig:kill-frap:frapinator}
      \end{sidewaysfigure}

      After we implemented the automatic identification of ROIs, the
      analysis script was turned into a self-contained % or stand-alone?
      program. Since interaction with the Octave interpreter is no
      longer required, a user does not have to ``program'' and can
      abstract from the language it is written. Options can be set via
      the command-line in a typical Unix style, and
      several images are saved with the results and all intermediate analysis
      in the form of visual logs for post-processing analysis.
      These visual logs include the automatically identified ROIs,
      as well multiple plots displaying the measured
      intensities, profile of the bleach spot, and fitting
      to different models \fref{fig:kill-frap:frapinator}.

      With this changes, human interaction was no longer required
      to perform the short time FRAP that the code was originally
      designed for.
      We have named this program frapinator, and made it available
      for public download under the GPL \footnote{\url{https://github.com/af-lab/frapinator}}.


  \subsection{Tracking of cell nuclei}

    %% We could show this but it would only be to "encher chouricos"
%    \begin{figure}
%      \centering
%      \missingfigure{Our first FRAP experiment}
%      \captionIntro{Long time series of HeLa cells expressing H2B--EGFP.}
%                   {Cells were transfected with pBOS--H2B-EGFP and imaged for 8
%                    hours, with intervals of 20 minutes.}
%      \label{fig:kill-frap:cell-movement}
%    \end{figure}

    Due to the extremely slow kinetics of histone proteins, FRAP
    must be performed over several hours where cell movement is
    an issue. % \fref{fig:kill-frap:cell-movement}.
    We tried different approaches to fix this.

    \subsubsection{Higher cell density}

      Initial attempt to limit movement of cells during FRAP
      was to perform the experiment with cells at higher confluence levels.
      Normal cells display contact inhibition, a cellular growth mechanism
      by which cells enter senescence and reduce motion, when
      surrounded by other cells with no free space for movement.

      \begin{figure}
        %% We are only showing one cell rather than the whole field of
        %% of view because otherwise it's hard to notice the movement of
        %% individual cells. If we do display everything, we cell many
        %% nuclei that seem like their movement is smaller. If we do
        %% show it, we comment that we are unsure whether the movement
        %% is cellular or only nuclear.
        \centering
        \includegraphics[width=\textwidth]{kill-frap/confluent-hela.png}
        \captionIntro{Movement of confluent HeLa cells during FRAP experiment}
          {
            Cells reached confluence before the start of the
            experiment in an attempt to reduce motion. Instead, this caused
            cell nuclei to undergo heavy reshape as the cell apparently
            squeezes in between its neighbours. Half-nuclear FRAP performed in
            a confocal microscope over an interval of 8~hours. Top left panel
            is the pre-bleach image, while the others have a time-interval of
            21~minutes. Cells are a stable line derived from HeLa, expressing
            the H4~R45H mutant tagged with YFP.
          }
        \label{fig:kill-frap:confluent-hela}
      \end{figure}

      Cancerous cell lines usually lose this property. However, the reduced
      space should place some restriction in the movement and indeed, we
      observed some decrease of movement but not a complete immobilization.
      Tracking of the ROIs over time was still required \fref{fig:kill-frap:confluent-hela}.

    \subsubsection{Primary cell lines}

      Since HeLa cells, being a cancerous cell line that had lost
      the mechanism through which contact
      inhibition is activated, we experimented with a primary horse
      fibroblast cell line. Using this cell line we were capable to maintain
      a layer of healthy cells covering a Petri dish for several after
      reaching confluence.
      %% cell growth was halted instead of becoming over-confluent

      Transfection of primary cell lines have typically a much lower
      efficiency rate. In addition, we need to image cells after they
      reached confluence, and entered senescence, but transfection
      methods are more efficient when cells are actively dividing
      and lower expression with each cell division. As a compromise,
      we transfected cells \SI{70}{\percent} and imaged after 3 days.

      \begin{figure}
        \centering
        \includegraphics[width=\textwidth]{kill-frap/confluent-horse.png}
        \captionIntro{Movement of confluent primary cells during FRAP experiment}
          {
            Primary horse fibrolasts display contact inhibition and halt growth
            once they reach confluence. However, this does not stop cell
            motion which can still be seen moving. In addition, when compared
            to the cancer cell line HeLa (\fref{fig:kill-frap:confluent-hela}),
            the horse fibroblasts frequently rotated around the $x$ and $y$
            axis. Circle FRAP was performed in a widefield microscope.
            Top left panel is the pre-bleach image, while the others have a
            time-interval of 15~minutes. Cells were transiently transfected
            and are expressing H2B type1-J tagged with EGFP.
          }
        \label{fig:kill-frap:confluent-horse}
      \end{figure}

      However, even after reaching confluence, we observed movement of
      transfected cells \fref{fig:kill-frap:confluent-horse}. Actually,
      the nature of the observed movement was dramatically different from
      the one observed in HeLa cells. All of the transfected cells displayed
      a helix-like movement around the vector of their movement.
      In contrast, rotation of HeLa nuclei was mostly restricted to the
      $z$ axis.

    \subsubsection{CropReg}

      As an alternative to modify the cells behaviour, we implemented a program
      to perform cell tracking. We named this program CropReg
      after its work flow of consecutive image cropping and image registration.

      Firstly, the nuclei of interest is tracked by template-based matching
      by normalized cross-correlation. The nuclei to track
      is identified on the first frame and is used as template against the
      image on the following frame. For increased performance and robustness,
      instead of performing the operation against the entire next frame, only
      the region surrounding its original position is used. Sequential
      usage of this method created a stack of smaller images centred on
      the nuclei of interest.

      Since this was a missing feature in GNU Octave, it was implemented
      upstream, as the ``coeff'' option for scaling in the
      \texttt{xcorr2} function and released as part of the Octave Forge
      signal package version 1.2.0.

      %% TODO since there's more than one way to actually do the normalization,
      %% it might be a good idea to write down the actual math formula

      Secondly, to correct for rotational movement around the $z$ axis, the
      frames were aligned using rigid body geometric transformation from the
      previously released ImageJ plugin StackReg \citep{stackreg}.

      \begin{figure}
        \centering
        \includegraphics[width=\textwidth]{kill-frap/cropreg.png}
        %% imaging was done every 10 minutes, but we are skipping
        %% every other panel
        \captionIntro{Automatic tracking and alignment of moving cells}
          {
            Using CropReg, we successfully tracked individual cells during
            a time-series microscope experiment. The top left corner of each
            panel displays the tracked and aligned cell. Imaging was performed
            in a widefield microscope. Time interval between panels 20~minutes.
            Cells are a stable line derived from HeLa, expressing H3 tagged
            with YFP.
          }
        \label{fig:kill-frap:cropreg}
      \end{figure}

      Using this methodology, we were able to track individual cell nuclei
      throughout the entire FRAP experiments \fref{fig:kill-frap:cropreg} provided
      that nuclei from different cells did not overlap. This was a rare occurrence.

  \subsection{Chromatin movement}

    While performing the FRAP experiments, we observed some movement
    within the cell nuclei. These could not be accounted for simple rotational
    movement around the $x$ or $y$ axis, and resembled more the movement
    of individual bodies within the nuclei.

    \subsubsection{Selection of \G1{} cells}
      %% There's no chemical equilibrium in S phase

      A possible cause of this chromatin movement comes from changes in
      the cell cycle phase. During the S~phase, the DNA is replicated,
      doubling the content of the chromatin.
      More importantly, this breaks
      a core assumption of FRAP, that the system remains in equilibrium
      during the entire experiment. This does not hold if the DNA, the
      binding sites for our model, duplicate in number.

      If the FRAP experiments can't be performed during S~phase and
      mitosis, we are limited to \G1{} and \G2{}. Considering
      the length of the HeLa cell cycle and the requirements to image
      for a time period of 8~hours, we are further limited to \G1{}.
      In addition, the FRAP experiment must be performed early in
      \G1{}~phase to avoid crossing over to the S~phase.

      %% The only reason this was required was because the LSM 510
      %% that we were using could not make Z stack and time lapse
      %% at the same time.
%      \begin{figure}
%        \centering
%        \missingfigure{Hela cells splitting}
%        \captionIntro{Picking cells at early G$_1$.}
%                     {We imaged cells that were entering mitosis and picked their
%                      daughter cells for the FRAP experiments. Because HeLa cells lift
%                      away from the dish during mitosis, opening the
%                      pinhole and set the Z-center in between the cell dividing plane
%                      and dish bottom was necessary. Ends up nothing being properly in focus but we
%                      can track things fine. Of course, some cells still floated away.}
%        %% TODO explicit parameters
%        \label{fig:kill-frap:picking-early-g1}
%      \end{figure}

      To do this, cells in mitosis were selected and tracked during 4~hours.
      After this time period, we used the daughter cells which we could be
      confident of being in early \G1{}.
      %% we also waited some 2 hours after mitosis since that's when cells
      %% unpack their chromosomes.

      During mitosis, HeLa cells form a sphere slightly above the plane of
      other cells, and keep a weak connection to the growth surface.
      Because of this, they easily detach, which is the basis for the
      mitotic shake-off method, and float away from the field of vision
      which requires a larger number
      of initial selected cells. In addition, to minimize any effect that
      may arise from imaging, it was done at minimal laser power and every
      30~minutes, just enough to allow manual tracking.
      Finally, since our system did not permit simultaneous Z-stack and time
      lapse imaging, and cells in mitosis are in a separate focal plane,
      imaging was performed with the pinhole sized to the max and focused
      in between the two planes. While this
      created very blurred images, it allowed to visualize all cells during
      the entire procedure.

      However, even after selecting cells in this cell cycle, movement within
      the bleach spot could still be observed.

    \subsubsection{Inverse FRAP}

      Due to the non-homogeneous nature of the chromatin, it was difficult
      to assess the total extent of the observed movement. To
      better visualize this, we performed inverse FRAP which allows us
      to track the movement of the bleach spot only.

      For this purpose, we replaced the EGFP tag in our H2B plasmid
      with photoactivatable GFP (PAGFP), a GFP derivative that requires
      activation by a specific wavelength to become fluorescent. This
      allows us to activate a specific spot of the nucleus and visualize
      its movement.

      Since PAGFP cannot be easily detected before photoactivation, cells
      were co-transfected with mCherry--\textalpha--tubulin which localises
      exclusively to the cytoplasm, giving an outline of the nuclear region
      \fref{fig:kill-frap:ifrap}.

      \begin{figure}
        \centering
        \subbottom[pre-activation]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/ifrap-pre.png}
          \label{fig:kill-frap:ifrap-pre}
        }
        \hfill
        \subbottom[post-activation]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/ifrap-post.png}
          \label{fig:kill-frap:ifrap-post}
        }
        \subbottom[activated spot over time]{
          \includegraphics[width=\textwidth]
          {kill-frap/ifrap.png}
          \label{fig:kill-frap:ifrap-timeframe}
        }
        \captionIntro{Inverse FRAP experiment showing chromatin movement}
          {
            HeLa cells co-transfected with mCherry--\textalpha--tubulin and
            H2B type1-J tagged with PAGFP.
            \subcaptionref{fig:kill-frap:ifrap-pre} The cell nucleus, target
            for photoactivation, can be easily identified as the ``empty''
            region via the mCherry channel on which would otherwise be an
            invisible feature on the GFP channel;
            \subcaptionref{fig:kill-frap:ifrap-post} spot after activation;
            \subcaptionref{fig:kill-frap:ifrap-timeframe} detail of the
            activated spot every 20~minutes. Rather than a gradual loss of
            fluorescence that maintains the circular shape, the activated spot
            kind of unfolds itself spreading the region of interest.
          }
        \label{fig:kill-frap:ifrap}
      \end{figure}

      Using this FRAP variant, the movement of chromatin was more noticeable.
      Rather than an homogeneous loss of fluorescence, the activated
      spot uncurled itself overtime with individual branches of
      localized PA-GFP appearing in the nuclei (\fref{fig:kill-frap:ifrap}).

