\chapter{Introduction}
\label{ch:intro}

\epigraph{You need a David filter or you'll go crazy.}{Holger Stephan}
%% because this is the chapter where I'll talk a lot
%%
%% "Don't indulge in sesquipedalian lexicological constructions" William Safire's Rules for Writers
%%
%% "A month in the laboratory can often save an hour in the library" F. H. Westheimer
%%
%% "Science is what we understand well enough to explain to a computer. Art is everything else we do."
%% by Donald E. Knuth (foreword to “A=B” by Petkovsek, W and Z)
%%
%% "By relieving the brain of all unnecessary work, a good notation sets it free to
%% concentrate on more advanced problems, and, in effect, increases the mental power
%% of the race." Quoted in P. Davis and R. Hersh The Mathematical Experience (Boston 1981)
%%
%% "The gall of them, fighting back!" -- Tyrion Lannister

%% FIXME 2nd year report abstract
%%The nucleosome is the fundamental structural unit of chromatin and the foundation
%%of it dynamics. Certain mutants have been shown to have a large effect on the
%%stability of the nucleosome \textit{in vitro} and to mimic chromatin remodelling
%%complexes in yeast.
%%
%%Fluorescence Recovery After Photobleaching (FRAP) has been previously used to
%%measure kinetics of endogenous histones. We are attempting to use it to measure
%%differences between wild type histones and mutants showing unusual \textit{in vitro} dynamics.
%%
%%We have developed two programs on the GNU Octave programming language, CropReg and
%%FRAPINATOR which was released under GPL. CropReg is able to track the nuclei of moving
%%cells. FRAPINATOR is able to do all the image processing, data extraction and fitting
%%of our FRAP experiments. These have revealed significant challenges to the determination
%%of relevant kinetic parameters, principally due to movement of chromatin.

\section{Chromatin}

  %% This estimate ignores mitochondrial DNA (but at this scale, its size
  %% is negligible anyway).
  The human genome has been estimated to be about \num{3.2e9}
  base pairs long \citep{nature-first-human-genome-draft}, which
  being an estimate for an haploid genome, means that the human cell will
  typically have \SI{6.4}{\giga\bp} worth of genomic information inside
  its nucleus.
  All of this is organised in a large DNA--protein complex known as chromatin.

  In chromatin, access to the genomic information is not all equal.
  There is an hierarchical structure with
  multiple levels of compaction
  that are achieved by folding the previous level into higher order structures
  that eventually form a chromosome.
  As the compaction level increases, access to the underlying genomic
  information decreases.
  The most simplistic and common view on the subject
  identifies two levels: euchromatin and heterochromatin, for low and
  high compaction respectively.
  This binary distinction is associated with active and inactive genes
  since high level of compaction prevents DNA transcription by blocking
  access to the all the required machinery.

  However, chromatin function is not limited to the compaction of DNA
  to fit on the
  cell nucleus, although it is often seen as so.
  Chromatin controls access
  to the DNA which also means controlling transcription and replication, as
  well as protecting DNA from damage;  it is about controlled and efficient
  access to genomic information, and not about a barrier problem for the sake
  of compaction.

  \subsection{Nucleosome structure}

    \begin{figure}
      \centering
      \missingfigure{subfigures with typical scheme of a nucleosome, side
        by side with 3d structure structure.  Show histone fold domain by making
        figures for each pair.}
      \caption{people usually draw as a tuna can. Here's how it really
        looks like.}
      \label{fig:intro:nucleosome-structure}
    \end{figure}

    Histone Fold Domain

    histone fold complex

  \subsection{Chromatin as substrate}

  \subsection{Chromatin remodelling}

\section{Centromeres --- specialized chromatin}
  \subsection{Centromere and kinetochore structure}
    \subsubsection{Timing of assembly}
  \subsection{Histone Fold Domain}
    \subsubsection{CENP--T/W}
    \subsubsection{CENP--S/X}

\section{Fluorescence microscopy}

  Fluorescence microscopy is an essential tool in the field of life
  sciences for the visualization of cells and tissues.
  Fluoropores emit light in a specific wavelength after
  being exposed to another specific wavelength, named
  emission and excitation wavelength respectively.
  Illumination of a sample with the excitation wavelength, and filtering
  out all but the emission wavelength from the formed image, allows
  the localization of fluorophores in a dark background.
  If specific molecules and cells can be associated with a fluorophore,
  so can their localization can be infered, and by choosing fluorophores
  with different emission wavelengths multiple molecules can be
  simultaneously identified.
  It would not be possible in normal optical microscopy.

  \subsection{Fluorescent proteins}
    Several fluorescent labels have been developed with different
    properties, each best suited to different techniques.
    These are often split into two major classes: synthetic fluorescent
    molecules such as DAPI or Fluorescein, which either bind to the molecule of
    interest or can be chemically linked to it; and Fluorescent Proteins (FPs)
    which are genetically encodable and can be expressed by the cells themselves.
    FPs allow tagging of individual molecules inside live cell, allowing
    visualization of dynamics in real-time live-cell imaging.
    Synthetic molecules are added exogenously, which often requires cell
    fixation and permeabilization disturbing intracellular structures.
    %% TODO mention synthethic moleculus that bing to proteins with tag like snap
    %%      Over simplistic?  Fusion proteins for luciferase, etc

    The \species{Aequorea} jellyfish Green Fluorescent Protein (GFP) was
    the first FP to be expressed recombinantly.
    Although originally discovered in 1962 \citep{shimomura1962-gfp-discovery},
    it was only in 1994 that it was used as a gene expression
    marker, by being placed under the control of \gene{T7} and
    \gene{mec-7} promoters, in \species{E. coli} and \species{C. elegans}
    respectively \citep{gfp-first-expression-marker}.
    %% when was GFP first used as a gene tag?

    Since then, many GFP variants have been engineered by mutating its
    original nucleotide sequence.  These not only increase fluorescence
    quantum yield, photostability, and improved folding, but also change
    the excitation and emission spectrum, providing a wide range of GFP
    derivatives with different colours.  FPs in other organisms were also
    discovered and were themselves engineered to provide further improved
    variants \citep{FP-color-palette}.

    An important class of FPs that have since been developed is the
    photo-controllable FPs whose
    fluorescent properties can be controlled by excitation with specific
    wavelengths allowing for individual cells and proteins to be optically
    labeled making this proteins specially useful for studies of cell
    lineage and protein movement.
    They are phenotypically split into three categories:

    \begin{description}
      \item[photo-activatable]
        Irreversible dark to bright state conversion.
        PA--GFP (Photo Activatable GFP), the first PA--FP to be
        reported, was developed by mutating the original GFP
        \citep{pagfp-discovery} and is
        still the only green PA--FP.
        It allows to mark and track a subset of selected molecules
        within cells.

      \item[photo-switchable]
        Irreversible conversion from one bright state to another bright
        state with a different emission wavelength.
        Like PA--FPs, they allow the tracking of a subset of selected
        molecules with the added advantages that the whole set is visible
        before the switch, and that the non-switched molecules continue
        to be visible.
        The development of proteins from this group started with Kaede FP
        \footnote{Kaede is named after maple leaf in Japanese but it was
        actually cloned from \species{Trachyphyllia geoffroyi} which is a
        stony coral.} whose wild type form is an obligate homotetrameric
        complex \citep{kaede-discovery}, thus making it
        unsuitable for use as a genetically encoded fusion tag.
        Monomeric Kaede-like FP have since been developed such as
        mEos2 \citep{meos2-discovery}.

      \item[reversible photo-switchable]
        Reversible conversion between dark and bright states, \eg Dronpa.
        These FPs have been mostly used in single molecule localization
        microscopy, a type of super-resolution microscopy, due to the
        low energy required for the transition between states.
        %% We could mention RESOLFT but that would be too much detail.

    \end{description}


  \subsection{FRAP}

    Fluorescence Recovery After Photobleaching (FRAP) is an optical
    microscopy technique to measure the dynamics of fluorescently
    tagged molecules.
    Originally developed in the 1970's for quantitative dynamics of lipids
    in cell membrane under the name Fluorescence Photobleaching
    Recovery (FPR) \citep{axelrod1976mobility}, it has
    been extensively used to obtain qualitative and quantitative
    insight on the kinetic properties of proteins since the development
    of GFP tagging.
    From studies of X, Y, and Z, to I, J, and K, FRAP is an ubiquitous
    technique in the field of cell biology with continual reviews being
    published\addref{for all the reviews about it every year}.

    In this technique, fluorescently tagged molecules in a small region
    are photobleached.  The fluorescent intensity in the region is
    measured over time to obtain a recovery curve.  The recovery is a
    function of the molecule dynamics, it represents the unbleached
    molecules, from outside the bleached region, moving into the
    photobleached region.  The molecule dynamics is determined by the
    transport and diffusion rates, as well as binding interactions.

    In the simplest case, a FRAP recovery curve is analysed
    qualitatively.  By looking at the plotted data, one assesses
    whether recovery is fast or slow, presence of an immobile
    fraction and binding interactions, and how this changes
    between different cases \fref{fig:intro:frap-curve-example}.

    \begin{figure}
      \centering
      \missingfigure{shwo recovery curve example, with parts.}
      \caption{}
      \label{fig:intro:frap-curve-example}
    \end{figure}

    In quantitate FRAP, data analysis involves fitting the recovery
    data to an idealized mathematical model.
    Many different models exist with different assumptions for the model
    \fref{fig:intro:frap-model-components}, and dependent on different
    parameters.
    Kinetic parameters are typically \Kon{} and \Koff{}, for
    binding and unbinding rate constants, and $D_{f}$ for diffusion constant.
    Different parameter values for the model are tested, until a curve is
    obtained that fits best to the measured recovery.

    \begin{figure}
      \centering
      \missingfigure{table of different assumptions as in Macnally's 2010 review}
      \caption{blah blah blah, there's so many models, how can anyone trust
               any of this?}
      \label{fig:intro:frap-model-components}
    \end{figure}

    Two factors contribute for the widespread use of this technique:
    it doesn't require specialized instruments and can be performed with
    any standard confocal microscope or widefield microscope with a laser
    module; and recovery curves are easy to understand making it easy to
    identify artifacts.  This is not the case with alternative techniques
    such as FCS (Fluorescence Correlation Microscopy) or single molecule
    tracking.

    The technique however has implicit assumptions which are applicable
    for most biological situations:

    \begin{itemize}
      \item The biological system has reached equilibrium and the equilibrium
        is maintained throughout the experiment, \ie{} the kinetic parameters
        must remain constant throughout the experiment;

      \item distribution of tagged protein mimics the endogenous protein;

      \item the binding sites are part of a large, relatively immobile
        complex, on the time and length scale of the recovery.
    \end{itemize}

    These assumptions become difficult to maintain as the experiment
    times increases, a requirement for slow moving molecules.  This
    is discussed again in \Cref{ch:kill-frap}.

  \subsection{iFRAP}


  \subsection{FLAP}

