\chapter{Introduction}

\section{Chromatin}

  %% This estimate ignores mitochondrial DNA but at this scale its size
  %% is negligible anyway, and we are talking about fitting it into the
  %% nucleus anyway.
  The human genome has been estimated to be about \num{3.2e9}
  base pairs long \citep{nature-first-human-genome-draft}, which
  being an estimate for an haploid genome, means that the human cell will
  typically have \SI{6.4}{\giga\bp} worth of genomic information inside
  its nucleus.
  All of this is organised in a large DNA--protein complex named chromatin.

  In chromatin, access to the genomic information is not all equal.
  It is organized in hierarchical structure with
  multiple levels of compaction \frefp{fig:intro:chromatin-structure} and
  as the level of compaction increases, access to the underlying genomic
  information decreases.
  The most simplistic and common view on the subject
  identifies two levels: euchromatin and heterochromatin, for low and
  high compaction respectively.
  This binary distinction is associated with active and inactive genes
  since high level of compaction prevents DNA transcription by blocking
  the access of the required machinery.

  However, chromatin function is not limited to the compaction of DNA
  so it fits on the
  cell nucleus, and hampered access to the DNA is not a negative side
  effect although it is often portrayed as such.
  Chromatin controls access
  to the DNA which means controlling transcription, replication,
  and recombination, as
  well as protecting DNA from damage;  it is about controlled and efficient
  access to genomic information and not about a barrier problem for the sake
  of compaction.

  In the chromatin hierarchical structure, the nucleosome is the
  basic repeating unit.
  An array of nucleosomes is then folded into a chromatin fiber by
  short-range interactions between neighbouring nucleosomes,
  and the subsequent interactions between chromatin fibers shape the
  condensed chromosome structure \frefp{fig:intro:chromatin-structure}.
  Or so is in the traditionally accepted model.  While the nucleosome
  structure is well established, details on the structure of the
  intermediary structures are still controversial and some recent models
  eschew them altogether
  \citep{fussner2011-no-30nm-fibre, luger2012-chromatin-review}.

  \begin{figure}
    \centering
    \def\svgwidth{\textwidth}
    \subinputfrom{figures/}{chromatin-hierarchy-diagram.pdf_tex}
    \captionIntro{Overview of the chromatin hierarchical structure.}%
                 {At the lowest level are the nucleosomes around which
                  DNA is wrapped.
                  This basic unit is repeated multiple times to form
                  the classical view of ``beads on a string''
                  \frefp{fig:intro:beads-on-a-string},
                  which is folded on itself to form the
                  \SI{30}{\nano\meter} fibre,
                  which is also folded on itself to form higher
                  compaction fibres until it finally forms the
                  mitotic chromosome.
                  Figure adapted from \cite{alberts} and \cite{lodish}.}
    \label{fig:intro:chromatin-structure}
  \end{figure}

  \subsection{Nucleosome}

    At the lowest level of chromatin compaction is the nucleosome.
    Each one is composed of the nucleosome core particle ---
    \SI{147}{\bp} of DNA wrapped around
    an histone protein octamer --- and the linker DNA ---
    a stretch of variable length DNA which connects
    to the following nucleosome.  Altogether, the nucleosome forms
    a repeating unit that includes \SIrange{165}{245}{\bp}
    \citep{widom1992-linker-length}.

    The first view of the nucleosome under a microscope revealed the
    classic beads on a string \footnote{originally, ``particles on a string''}
    structure of chromatin fibers
    \frefp{fig:intro:beads-on-a-string} \citep{olin1974-nu-bodies},
    establishing its position as the repeating subunit of chromatin.
    These results were a surprise as it departed from the then current view
    of chromatin structure as a uniform superhelical coiling of the
    DNA itself, stabilised by histone interactions
    \citep{pardon-wilkins-1972model}.

    Associated with the discovery of interactions between histones,
    first between H2A and H2B \citep{anna-isenberg-1974-h2a-h2b}, and
    then between H3 and H4 \citep{kornberg1974-results};
    the previous knowledge of the main histones being present in
    roughly equimolar concentrations with four histones per
    \SI{100}{\bp} of DNA \addref{where is this?};
    %% They didn't actually refer to 200bp on this paper, there is
    %% no ladder on the figure.  It's Kornberg on the notes of his
    %% references that mentions:
    %%    "the extent of digestion and size of the pieces are
    %%    from D. R. Hewish, personal communication
    %%    (the size was determined by velocity sedimenta-
    %%    tion in alkali and should be regarded as only
    %%    approximate)."
    and digestion of chromating by nucleases yielding
    \SI{200}{\bp} fragments \citep{hewish1973-200bp-pieces};
    led to the basic model of the nucleosome as is known today:

    \begin{quotation}
      Chromatin structure is based on a repeating unit of eight
      histone molecules and about 200 DNA base pairs.
      \sourceatright{\cite{kornberg1974-model}}
    \end{quotation}

    \begin{figure}
      \centering
      %% The original image has 3 panels:
      %%    a) rat thymus positive staining;
      %%    b) rat thymus negative staining;
      %%    c) chicken erythrocyte positive staining.
      %% We don't need to show all of them, so we show only c) because
      %% it's the nicest looking one.  We crop the other two panels out
      %% and a bit more too to remove the "C".  We turn it 90 degrees
      %% to save space too.
      \includegraphics[trim=24cm 0 0 0, clip=true, angle=-90, width=\textwidth]%
      {"figures/beads-on-a-string"}
      \label{fig:intro:beads-on-a-string}
      \captionIntro{Classic ``beads on a string'' view of chromatin.}%
                   {Chicken erythrocyte chromatin negatively stained.
                    Chromatin fibers are seen spilling out of ruptured
                    nuclei depicting the classic ``beads on a string''.
                    Scale bars of \SI{0.2}{\micro\meter}.
                    First published report on
                    visualization of nucleosomes, originally named
                    \textnu~bodies \citep{olin1974-nu-bodies}.}
    \end{figure}

    More details on the structure of the nucleosome core particle required
    solving its crystal structure.  Solving the structure by X-ray
    diffraction was initially reported in 1977
    \citep{finch1977-first-structure}
    with a low resolution until a reaching an
    intermediate resolution of
    revealing a disk shaped (H3--H4)\textsubscript{2} symmetric tetramer
    at the center of the nucleosome, two H2A--H2B dimers,
    one on each face of the disk, and the DNA sequence around it
    making several interactions with the histones octamer within
    \SI{7}{\angstrom} model \citep{richmond1984-7angstrom}.
    High resolutions structure was solved later
    \citep{arents1991-31angstrom,luger1997-28angstrom}
    with \SI{3.1}{\angstrom} and \SI{2.8}{\angstrom}
    respectively
    first showing the individual histone fold, named histone fold domain,
    and the ``handshake'' motif between the assembled histone dimers;
    and later the atomic interactions between the histones and their
    internal structure.

    A list of some of the
    different nucleosome crystal structures solved to date along with their histone species
    composition and DNA sequence is given in \addref{Tan and Davey, 2011}.

  \todo[inline]{Can we add section about histones on the introduction?  That is
        already the whole chapter of histone catalogue.  Maybe not since
        that is not required knowledge for the results chapter.
        Maybe make a reference that histones are better explained in
        that chapter?}

    \subsubsection{Histone Fold Domain}

      %% \textalpha1--L1--\textalpha2--L2--\textalpha3
      Core histone proteins share a common structural domain named
      histone fold, which consists of a long central
      \textalpha-helix flanked on both sides by a loop
      and shorter \textalpha-helices \citep{arents1991-31angstrom, arents1995histone-fold}.
      This domain allows for the characteristic ``handshake'' motif
      of dimer assembly where the central \textalpha-helix of each
      histone crosses each other and fits between the shorter
      \textalpha-helices of the other \frefp{fig:intro:histone-dimers-structure}.
      It gives histones an extensive molecular contact interface
      between each other.

      %% We could also write how interesting the histone fold is from an
      %% evolutionary point of view.  See Arents and Moudrianakis 1995.

      In addition, each histone have extra \textalpha-helix, \textbeta-strands,
      and unstructured N-- and C-- terminals \addref{add a figure with secondary structure}.

      \begin{figure}
        %% Two possible configurations:
        %%
        %%  This has a nice hierarchy that matches the nucleosome structure
        %%  so it's obvious to see the levels.
        %%  +-------+-------+-------+-------+
        %%  |       |       |       |       |
        %%  |  H2A  |  H2B  |   H3  |   H4  |
        %%  |       |       |       |       |
        %%  +-------+-------+-------+-------+
        %%  |    H2A-H2B    |     H3-H4     |
        %%  |     dimer     |     dimer     |
        %%  +---------------+---------------+
        %%  |                               |
        %%  |          nucleosome           |
        %%  |                               |
        %%  +-------------------------------+
        %%
        %%  This has the dimer on the same size as the individual histones
        %%  and makes it easier to see where each fits.
        %%  +--------+--------+-------------+
        %%  |        |        |             |
        %%  |   H2A  |   H2B  |   H2A-H2B   |
        %%  |        |        |    dimer    |
        %%  +--------+--------+-------------+
        %%  |        |        |             |
        %%  |    H3  |   H4   |    H3-H4    |
        %%  |        |        |    dimer    |
        %%  +--------+--------+-------------+
        %%  |                               |
        %%  |          nucleosome           |
        %%  |                               |
        %%  +-------------------------------+
        \centering
        \subbottom[]{%
          \missingfigure{secondary structure of histones}
          \label{fig:intro:histone-secondary-structure}
        }
        \subbottom[]{%
          \includegraphics[width=0.32\textwidth]%
          {"results/nucleosome-h2a"}
          \includegraphics[width=0.32\textwidth]%
          {"results/nucleosome-h2b"}
          \includegraphics[width=0.32\textwidth]%
          {"results/nucleosome-h2a-h2b-dimer"}
          \label{fig:intro:individual-histone-structure}
        }
        \subbottom[]{%
          \label{fig:intro:histone-dimers-structure}
        }
        \subbottom[]{%
          \missingfigure{nucleosome structure}
          \label{fig:intro:nucleosome-structure}
        }
        \captionIntro{Structure of the nucleosome core particle and constituents.}%
                     {lalala}
      \end{figure}

    \subsubsection{Histone tails}

      Each one of the four core histones has an unstructured N or C
      terminal tail that extend beyond the DNA in the nucleosome core,
      making the inter-nucleosome interactions and therefore an important
      role on chromatin higher order structures.
      In addition, the tails are subject to a wide range of
      post-translational modifications
      including acetylation, methylation, ubiquitination, phosphorylation,
      and ADP--ribosylation \citep{bannister2011ptm-review}.
      The vast number of post-translational modifications has led to the
      histone code hypothesis that these modifications are the basis for
      regulation of chromatin-templated processes \citep{jenuwein200histone-code}.

      %% Should we mention that there are a few PTM in the nucleosome
      %% core, it's just that the vast majority are on the tails?

      Histone post-translational modification have indeed been shown to
      regulate chromatin structure, not only by merely being there, but
      also by recruiting different protein complexes.
      They have been show to influence
      DNA processes such as DNA replication, repair, recombination,
      and transcription.
      The most studied modifications have been related with transcription,
      gene activation and gene silencing were even the only points
      on the original proposal for the histone code \citep{jenuwein200histone-code}
      reducing chromatin structure to an either ``on'' or ``off' state.
      For example,
      H3 K4 di-methylation and H3 K27 tri-methylation are well known
      markers for gene activation and silencing respectively.
      An example of non
      Phosphorilation of Serine 139 H2AX is involved in DNA repair
      in the recruitment of DNA repair-related proteins \Srefp{app:h2ax-review}.

    \subsubsection{Core histones}
    
      They will be discussed with more details on \Sref{section:histone-catalogue}.


\section{Fluorescence microscopy}

  Fluorescence microscopy is an essential tool in the field of life
  sciences for the visualization of cells and tissues.
  Fluoropores emit light in a specific wavelength after
  being exposed to another specific wavelength, named
  emission and excitation wavelength respectively.
  Illumination of a sample with the excitation wavelength, and filtering
  out all but the emission wavelength from the formed image, allows
  the localization of fluorophores in a dark background.
  If specific molecules and cells can be associated with a fluorophore,
  so can their localization can be infered, and by choosing fluorophores
  with different emission wavelengths multiple molecules can be
  simultaneously identified.
  It would not be possible in normal optical microscopy.

  \subsection{Fluorescent proteins}
    Several fluorescent labels have been developed with different
    properties, each best suited to different techniques.
    These are often split into two major classes: synthetic fluorescent
    molecules such as DAPI or Fluorescein, which either bind to the molecule of
    interest or can be chemically linked to it; and Fluorescent Proteins (FPs)
    which are genetically encodable and can be expressed by the cells themselves.
    FPs allow tagging of individual molecules inside live cell, allowing
    visualization of dynamics in real-time live-cell imaging.
    Synthetic molecules are added exogenously, which often requires cell
    fixation and permeabilization disturbing intracellular structures.
    %% TODO mention synthethic moleculus that bing to proteins with tag like snap
    %%      Over simplistic?  Fusion proteins for luciferase, etc

    The \species{Aequorea} jellyfish Green Fluorescent Protein (GFP) was
    the first FP to be expressed recombinantly.
    Although originally discovered in 1962 \citep{shimomura1962-gfp-discovery},
    it was only in 1994 that it was used as a gene expression
    marker, by being placed under the control of \gene{T7} and
    \gene{mec-7} promoters, in \species{E. coli} and \species{C. elegans}
    respectively \citep{gfp-first-expression-marker}.
    %% when was GFP first used as a gene tag?

    Since then, many GFP variants have been engineered by mutating its
    original nucleotide sequence.  These not only increase fluorescence
    quantum yield, photostability, and improved folding, but also change
    the excitation and emission spectrum, providing a wide range of GFP
    derivatives with different colours.  FPs in other organisms were also
    discovered and were themselves engineered to provide further improved
    variants \citep{FP-color-palette}.

    An important class of FPs that have since been developed is the
    photo-controllable FPs whose
    fluorescent properties can be controlled by excitation with specific
    wavelengths allowing for individual cells and proteins to be optically
    labeled making this proteins specially useful for studies of cell
    lineage and protein movement.
    They are phenotypically split into three categories:

    \begin{description}
      \item[photo-activatable]
        Irreversible dark to bright state conversion.
        PA--GFP (Photo Activatable GFP), the first PA--FP to be
        reported, was developed by mutating the original GFP
        \citep{pagfp-discovery} and is
        still the only green PA--FP.
        It allows to mark and track a subset of selected molecules
        within cells.

      \item[photo-switchable]
        Irreversible conversion from one bright state to another bright
        state with a different emission wavelength.
        Like PA--FPs, they allow the tracking of a subset of selected
        molecules with the added advantages that the whole set is visible
        before the switch, and that the non-switched molecules continue
        to be visible.
        The development of proteins from this group started with Kaede FP
        \footnote{Kaede is named after maple leaf in Japanese but it was
        actually cloned from \species{Trachyphyllia geoffroyi} which is a
        stony coral.} whose wild type form is an obligate homotetrameric
        complex \citep{kaede-discovery}, thus making it
        unsuitable for use as a genetically encoded fusion tag.
        Monomeric Kaede-like FP have since been developed such as
        mEos2 \citep{meos2-discovery}.

      \item[reversible photo-switchable]
        Reversible conversion between dark and bright states, \eg Dronpa.
        These FPs have been mostly used in single molecule localization
        microscopy, a type of super-resolution microscopy, due to the
        low energy required for the transition between states.
        %% We could mention RESOLFT but that would be too much detail.

    \end{description}


  \subsection{FRAP}

    Fluorescence Recovery After Photobleaching (FRAP) is an optical
    microscopy technique to measure the dynamics of fluorescently
    tagged molecules.
    Originally developed in the 1970's for quantitative dynamics of lipids
    in cell membrane under the name Fluorescence Photobleaching
    Recovery (FPR) \citep{axelrod1976mobility}, it has
    been extensively used to obtain qualitative and quantitative
    insight on the kinetic properties of proteins since the development
    of GFP tagging.
    From studies of X, Y, and Z, to I, J, and K, FRAP is an ubiquitous
    technique in the field of cell biology with continual reviews being
    published\addref{for all the reviews about it every year}.

    In this technique, fluorescently tagged molecules in a small region
    are photobleached.  The fluorescent intensity in the region is
    measured over time to obtain a recovery curve.  The recovery is a
    function of the molecule dynamics, it represents the unbleached
    molecules, from outside the bleached region, moving into the
    photobleached region.  The molecule dynamics is determined by the
    transport and diffusion rates, as well as binding interactions.

    In the simplest case, a FRAP recovery curve is analysed
    qualitatively.  By looking at the plotted data, one assesses
    whether recovery is fast or slow, presence of an immobile
    fraction and binding interactions, and how this changes
    between different cases \fref{fig:intro:frap-curve-example}.

    \begin{figure}
      \centering
      \missingfigure{shwo recovery curve example, with parts.}
      \caption{}
      \label{fig:intro:frap-curve-example}
    \end{figure}

    In quantitate FRAP, data analysis involves fitting the recovery
    data to an idealized mathematical model.
    Many different models exist with different assumptions for the model
    \fref{fig:intro:frap-model-components}, and dependent on different
    parameters.
    Kinetic parameters are typically \Kon{} and \Koff{}, for
    binding and unbinding rate constants, and $D_{f}$ for diffusion constant.
    Different parameter values for the model are tested, until a curve is
    obtained that fits best to the measured recovery.

    \begin{figure}
      \centering
      \missingfigure{table of different assumptions as in Macnally's 2010 review}
      \caption{blah blah blah, there's so many models, how can anyone trust
               any of this?}
      \label{fig:intro:frap-model-components}
    \end{figure}

    Two factors contribute for the widespread use of this technique:
    it doesn't require specialized instruments and can be performed with
    any standard confocal microscope or widefield microscope with a laser
    module; and recovery curves are easy to understand making it easy to
    identify artifacts.  This is not the case with alternative techniques
    such as FCS (Fluorescence Correlation Microscopy) or single molecule
    tracking.

    The technique however has implicit assumptions which are applicable
    for most biological situations:

    \begin{itemize}
      \item The biological system has reached equilibrium and the equilibrium
        is maintained throughout the experiment, \ie{} the kinetic parameters
        must remain constant throughout the experiment;

      \item distribution of tagged protein mimics the endogenous protein;

      \item the binding sites are part of a large, relatively immobile
        complex, on the time and length scale of the recovery.
    \end{itemize}

    These assumptions become difficult to maintain as the experiment
    times increases, a requirement for slow moving molecules.  This
    is discussed again in \Cref{ch:kill-frap}.

  \subsection{iFRAP}


  \subsection{FLAP}

