\section{Chromatin}

  %% Note that the size of an human genome is for an haploid genome.  The
  %% actual number of bp inside a typical cell will be double of that.
  %% Also, the estimate ignores mitochondrial DNA but at this scale its size
  %% is negligible anyway, and we are talking about fitting it into the
  %% nucleus anyway.
  The human genome has been estimated to be about \num{3.2e9}
  base pairs long \citep{nature-first-human-genome-draft}
  meaning that a human cell will
  typically have \SI{6.4}{\giga\bp} worth of genomic information inside
  its nucleus.
  All of this is organised in a large DNA--protein complex named chromatin.

  In chromatin, access to the genomic information is not all equal.
  It is organized in a hierarchical structure with
  multiple levels of compaction \frefp{fig:intro:chromatin-structure} where
  the level of compaction increases as access to the underlying genomic
  information decreases.
  The most simplistic and common view on the subject
  identifies two levels: euchromatin and heterochromatin for low and
  high compaction respectively.
  This binary distinction is associated with active and inactive genes
  since high level of compaction prevents DNA transcription by blocking
  the access of the required machinery.
  However, chromatin function is not limited to the compaction of DNA
  so it fits on the
  cell nucleus, and hampered access to the DNA is not a negative side
  effect although it is often portrayed as such.
  Chromatin controls access
  to the DNA which means controlling transcription, replication,
  and recombination, as
  well as protecting DNA from damage;  it is about controlled and efficient
  access to genomic information and not about a barrier problem for the sake
  of compaction \citep{controlling-double-helix}.

  \begin{figure}
    \centering
    \def\svgwidth{\textwidth}
    \subinputfrom{figures/}{chromatin-hierarchy-diagram.pdf_tex}
    \captionIntro{Overview of the chromatin hierarchical structure.}%
                 {At the lowest level are the nucleosomes around which
                  DNA is wrapped.
                  This basic unit is repeated multiple times to form
                  the classical view of ``beads on a string''
                  \frefp{fig:intro:beads-on-a-string},
                  which is folded on itself to form the
                  \SI{30}{\nano\meter} fibre,
                  which is also folded on itself to form higher
                  compaction fibres until it finally forms the
                  mitotic chromosome.
                  Figure adapted from \cite{alberts} and \cite{lodish}.}
    \label{fig:intro:chromatin-structure}
  \end{figure}

  \begin{figure}
    \centering
    %% The original image has 3 panels:
    %%    a) rat thymus positive staining;
    %%    b) rat thymus negative staining;
    %%    c) chicken erythrocyte positive staining.
    %% We don't need to show all of them, so we show only c) because
    %% it's the nicest looking one.  We crop the other two panels out
    %% and a bit more too to remove the "C".  We turn it 90 degrees
    %% to save space too.
    \begin{tikzpicture}
        \node[anchor=south west,inner sep=0] at (0,0)
          {\includegraphics[trim=24cm 0 0 0, clip=true, angle=-90,%
                            width=\textwidth]{"figures/beads-on-a-string"}};
        \node at (1.7,0.35)
          {\SI{0.2}{\micro\meter}};
    \end{tikzpicture}
    \captionIntro{Classic ``beads on a string'' view of chromatin.}%
                 {Chicken erythrocyte chromatin negatively stained.
                  Chromatin fibers are seen spilling out of ruptured
                  nuclei depicting the classic ``beads on a string''.
                  First published report on
                  visualization of nucleosomes, originally named
                  \textnu~bodies \citep{olins1974-nu-bodies}.}
    \label{fig:intro:beads-on-a-string}
  \end{figure}

  In the chromatin hierarchical structure, the nucleosome is the
  basic repeating unit.
  An array of nucleosomes is then folded into a chromatin fiber by
  short-range interactions between neighbouring nucleosomes
  and the subsequent interactions between chromatin fibers shape the
  condensed chromosome structure \frefp{fig:intro:chromatin-structure}.
  Or so is in the traditionally accepted model.  While the nucleosome
  structure is well established, details on the structure of the
  intermediary structures are still controversial and some recent models
  eschew them altogether
  \citep{fussner2011-no-30nm-fibre, luger2012-chromatin-review}.

  The nucleosome itself is composed of a core particle
  consisting of \SI{147}{\bp} of DNA and an histone protein octamer,
  and a variable length linker DNA that connects
  to the following core particles \frefp{fig:intro:nucleosome-top-view}.
  Altogether, the nucleosome includes \SIrange{165}{245}{\bp} of DNA
  \citep{widom1992-linker-length}.

  Nucleosomes act as the substrate for almost all nuclear processes that
  require access to genomic DNA \citep{controlling-double-helix}.
  Reconfigurations of local chromatin necessary for these processes can
  be accomplished by changing nucleosome structure, or altering its
  composition by incorporation of variant histones or post-translational
  modifications.
  Understanding chromatin structure and the nature of changes which can
  occur requires understanding the biochemical properties of nucleosomes
  \Srefp{sec:intro:chromatin-remodelling}.

  \subsection{Histones}

    Histones is a family of proteins, the principal protein component
    of chromatin, and the only proteins present in the nucleosome.
    There are five histone types: H1, H2A, H2B, H3, and H4.
    Histone H1 binds to the linker DNA and is not part of the nucleosome.
    The other four form the core
    nucleosome and are therefore referred to as core histones.
    Their highly basic charge has always been a hallmark characteristic of
    this proteins.  The DNA polymer is negatively charged and the highly
    basic histone proteins bind to it, neutralizing the negative charge
    and allowing for the required compaction of DNA in the nucleus.
    In addition, histones themselves interact with each other, forming
    a compact histone octamer structure that further helps with
    DNA compaction.

    The histone octamer is the central piece of the nucleosome core particle
    \frefp{fig:intro:nucleosome-top-view}
    composed of two of each of the core histones arranged in four dimers
    --- two H2A--H2B and two H3--H4.
    The two H3--H4 dimers form a disk-shaped (H3--H4)\textsubscript{2}
    tetramer at the center \frefp{fig:intro:tetramer-side-view}
    with one H2A--H2B dimer on each face of the disk
    \frefp{fig:intro:octamer-side-view}
    and the \SI{147}{\bp} of DNA wrapped around in 1.65~turns
    \frefp{fig:intro:nucleosome-side-view}.

    \begin{figure}
      \centering
      \subbottom[]{%
        \includegraphics[width=0.5\textwidth]%
        {"results/nucleosome-top"}
        \label{fig:intro:nucleosome-top-view}
      }

      %% We definitely want the following 3 images side-by-side
      \subbottom[Nucleosome core particle]{%
        \includegraphics[width=0.3\textwidth]%
        {"results/nucleosome-side"}
        \label{fig:intro:nucleosome-side-view}
      }
      \subbottom[Histone octamer]{%
        \includegraphics[width=0.3\textwidth]%
        {"results/octamer-side"}
        \label{fig:intro:octamer-side-view}
      }
      \subbottom[(H3--H4)\textsubscript{2} tetramer]{%
        \includegraphics[width=0.3\textwidth]%
        {"results/tetramer-side"}
        \label{fig:intro:tetramer-side-view}
      }
      \captionIntro{Structure of the nucleosome core particle.}%
                   {Axial view of the nucleosome core particle
                    facing towards the DNA entry/exit point
                    \subcaptionref{fig:intro:nucleosome-side-view}
                    showing 1.65 turns of DNA wrapped around the
                    histone octamer;
                    \subcaptionref{fig:intro:octamer-side-view}
                    DNA hidden showing the histone octamer;
                    \subcaptionref{fig:intro:tetramer-side-view}
                    the two H2A--H2B dimers hidden to show the
                    disk shaped (H3--H4)\textsubscript{2} tetramer center.
                    Figure generated from PDB structure 2CV5
                    \citep{tsunaka2005-2cv5}.
                    Histones H2A, H2B, H3, and H4 are coloured cyan,
                    grey, yellow, and blue respectively.}
    \end{figure}

    \subsubsection{Histone and nucleosome structure}

      The structure of the core histones proteins themselves, is divided
      into three parts: histone fold domain --- the central piece of an
      histone involved in the formation of histone dimers, the histone
      fold extensions, and the histone tails --- regions of the histone
      that extend beyond the nucleosome interacting with the DNA and
      other nucleosomes.

      %% Histone fold domain
      %% \textalpha1--L1--\textalpha2--L2--\textalpha3
      Core histone proteins share a common structural domain named
      histone fold, which consists of a long central
      \textalpha-helix flanked on both sides by a loop
      and shorter \textalpha-helices \citep{arents1991-31angstrom, arents1995histone-fold}.
      This domain allows for the characteristic ``handshake'' motif
      of dimer assembly where the central \textalpha-helix of each
      histone crosses each other and fits between the shorter
      \textalpha-helices of the other (\fref{fig:intro:h2a-h2b-structure}
      and \ref{fig:intro:h3-h4-structure}).
      It gives histones an extensive molecular contact interface
      between each other.

      \begin{figure}
        %% Two possible configurations:
        %%
        %%  This has a nice hierarchy that matches the nucleosome structure
        %%  so it's obvious to see the levels.
        %%  +-------+-------+-------+-------+
        %%  |       |       |       |       |
        %%  |  H2A  |  H2B  |   H3  |   H4  |
        %%  |       |       |       |       |
        %%  +-------+-------+-------+-------+
        %%  |    H2A-H2B    |     H3-H4     |
        %%  |     dimer     |     dimer     |
        %%  +---------------+---------------+
        %%
        %%  This has the dimer on the same size as the individual histones
        %%  and makes it easier to see where each fits.
        %%  +--------+--------+-------------+
        %%  |        |        |             |
        %%  |   H2A  |   H2B  |   H2A-H2B   |
        %%  |        |        |    dimer    |
        %%  +--------+--------+-------------+
        %%  |        |        |             |
        %%  |    H3  |   H4   |    H3-H4    |
        %%  |        |        |    dimer    |
        %%  +--------+--------+-------------+
        \centering
        \subbottom[Histone secondary structure]{%
          \missingfigure{secondary structure of histones}
          \label{fig:intro:histone-secondary-structure}
        }

        \subbottom[H2A]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h2a"}
          \label{fig:intro:h2a-structure}
        }
        \subbottom[H2B]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h2b"}
          \label{fig:intro:h2b-structure}
        }
        \subbottom[H2A--H2B dimer]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h2a-h2b-dimer"}
          \label{fig:intro:h2a-h2b-structure}
        }

        \subbottom[H3]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h3"}
          \label{fig:intro:h3-structure}
        }
        \subbottom[H4]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h4"}
          \label{fig:intro:h4-structure}
        }
        \subbottom[H3-H4 dimer]{%
          \includegraphics[angle=90, width=0.32\textwidth]%
          {"results/nucleosome-h3-h4-dimer"}
          \label{fig:intro:h3-h4-structure}
        }

        \captionIntro{Histone fold domain.}%
                     {lalala.
                      Figure generated from PDB structure 2CV5
                      \citep{tsunaka2005-2cv5}.
                      Histones H2A, H2B, H3, and H4 are coloured cyan,
                      grey, yellow, and blue respectively.}
      \end{figure}

      %% Histone tails

      Each one of the four core histones has an unstructured N or C
      terminal tail that extend beyond the DNA in the nucleosome core,
      making the inter-nucleosome interactions and therefore an important
      role on chromatin higher order structures.
      In addition, the tails are subject to a wide range of
      post-translational modifications
      including acetylation, methylation, ubiquitination, phosphorylation,
      and ADP--ribosylation \citep{bannister2011ptm-review}.
      The vast number of post-translational modifications has led to the
      histone code hypothesis that these modifications are the basis for
      regulation of chromatin-templated processes \citep{jenuwein200histone-code}.

      %% Should we mention that there are a few PTM in the nucleosome
      %% core, it's just that the vast majority are on the tails?

      Histone post-translational modification have indeed been shown to
      regulate chromatin structure, not only by merely being there, but
      also by recruiting different protein complexes.
      They have been show to influence
      DNA processes such as DNA replication, repair, recombination,
      and transcription.
      The most studied modifications have been related with transcription,
      gene activation and gene silencing were even the only points
      on the original proposal for the histone code \citep{jenuwein200histone-code}
      reducing chromatin structure to an either ``on'' or ``off' state.
      For example,
      H3 K4 di-methylation and H3 K27 tri-methylation are well known
      markers for gene activation and silencing respectively.
      An example of non
      Phosphorilation of Serine 139 H2AX is involved in DNA repair
      in the recruitment of DNA repair-related proteins \Srefp{app:h2ax-review}.

    \subsubsection{Histone variants}

      Each of the histones protein types is encoded by a large family
      of genes resulting in several slightly different proteins.
      These are grouped into two classes, canonical isoforms
      and variant histones, which are based on their gene location,
      expression characteristics, and functional roles.
      Canonical isoforms contribute to the majority
      of histones in chromatin, have high sequence identify, and display
      largely equivalent functions.  They are discussed in more detail on
      \Sref{section:histone-catalogue}.
      Variant histones are present in a smaller number than canonical
      histones, much different sequences with unique structural features,
      and perform a variety of specialized functions.  For example, the
      H2A variant H2AX has a longer C-terminal tail with unique locations
      for post-translational modifications involved in DNA damage
      response \Arefp{app:h2ax-review}.  CENP-A is a H3 variant that
      forms a more compact and rigid nucleosome and has a unique N-terminal
      tail that is involved in centromere formation \citep{black2011-cenpa}.

    \subsubsection{Histone H1}
      %% I guess we should at least mention a bit of H1.

      %% Luger 1997 specifically says that H1 is part of the nucleosome.
      %% Other older papers also include H1 as part of the nucleosome
      %% structure.  That doesn't happen so often anymore.
      Histone H1 is not part of the nucleosome structure and is not
      required for the beads on a string structure.
      Instead, it binds to the linker DNA, which is why it is referred
      to as linker histone, to stabilise the higher order
      chromatin structures and modulate the accessibility of
      regulatory proteins,
      chromatin remodelling factors, and histone modification enzymes
      to their target sites.
      As with the case of higher order chromatin structures, details
      on H1 contribution to chromatin structure and function remains
      unclear but the typically accepted view is that it binds at
      the nucleosome DNA entry/exit points and that it shortens the
      length of the linker DNA bringing nucleosomes close together
      to form the \SI{30}{\nm} fibres \citep{harshman2013h1-review}.

      Similar to the core histones, expression of histone H1 is
      replication dependent, there are variant H1 with specialised
      roles and tissue-specific expression, all canonical genes
      are located at an histone gene cluster, and the H1 protein
      is target to multiple post-translational modifications.
      Structuraly, histones H1 consist of a short N terminal, a globular
      structure, and a variable length C terminal with a high lysine
      content, which is why they are also named lysine-rich histones.

      There are fundamental differences in the functional roles of
      the core histones and histone H1;
      there is no sequence similarity between these groups;
      histone H1 does not have an histone fold;
      histone H1 is not evolutionary conserved like core histones;
      histone H1 is not required for nucleosome formation.
      Why, then, should histone H1 and core histones be even considered together?
      The answer seems to be a rather trivial one: 
      ``histon'' was the name given to the protein fraction of
      the original crude chromatin extracts \citep{kossel1884-histones}
      even though histone H1 was the first species to be separated from
      the others \citep{stedman1951main-histones-separation}.

    \subsubsection{Historical perspective}
      %% History of study of the structure and function of chromatin.

      The first chromatin preparation, at the time named nucleïn, was
      separated in two components: one basic and protein-like which
      became known as histones, and one acid and unlike any cellular
      substance yet observed which became known as nucleic acid.
      This discovery predates the discovery of DNA as the hereditary
      molecule and genetic material, the discovery of DNA itself,
      the discovery of the nucleus as the recipient for genetic
      information, and even the discovery of proteins as polypeptides.
      Actually, this discovery was a precursor for all those.

      From the initial chromatin preparation to the crystal structure of
      the nucleosome at high resolution, varied models for chromatin
      existed.  Even the nucleosome itself was discovered more than 100 years
      after the chromatin preparations.
      The series of discoveries that led to the model of nucleosome
      structure is better told in a timeline fashion with references
      to the original reports.

      \begin{description}
        %% Ueber die chemische Zusammensetzung der Eiterzellen
        %% von Dr. F. Miescher aus Basel
        \item[\cite{miescher1871-chromatin}]
        First chromatin preparation, named nucleïn.

        \item[\cite{miescher1874-protamines}]
        First DNA purification, also named nucleïn, and identification
        of protamines from a sperm chromatin preparation.  Demonstration
        that DNA exhibits acidic properties.

        \item[\cite{flemming1880-chromatin}]
        Chromatin named for the first time as the readily stainable
        material on cell nuclei. Only later would it be identified
        as the same as the nucleïn in Miescher preparations.

        %% Wenn in dem vorliegenden Fall ebenfalls eine derartige Verbindung
        %% des Nucleïns vorlag, so war eine Extraktion des basischen Körpers
        %% durch die verdünnte Säure vorauszusetzen. Die Untersuchung des
        %% salzsauren Extraktes der Kernmasse ergab die Anwesenheit eines
        %% Körpers, welcher zu jener Klasse von Substanzen gehört, due unter
        %% dem Namem A-Pepton (Meissner), Propepton (Schmidt-Mühlheim),
        %% Albumose (Kühne) zusammengefasst werden. [...]
        %% Ich schlage für diese Substanz den Namen Histon vor.
        \item[\cite{kossel1884-histones}]
        Histones named for the first time.  They were the peptone-like
        substances with basic properties obtained by acid extraction from
        a chromatin preparation.

        %% Thanks to people on ##deutsch on irc.freenode.net for
        %% help with the translation.
        %%
        %% Die Bindung des Histons am Nuclein muß wahrscheinlich als
        %% eine saltzartige ausgefaßt werden und zwar so, daß die
        %% Säurekomponente durch das Nuclein, der basische Bestandteil
        %% durch das Histon vertreten ist.
        %%
        %% saltzartige --> salt-like --> electrostatic
        \item[\cite{huiskamp1903-electrostatic}]
        After a series of studies on chromatin chemical properties,
        including demonstration that nucleïn is negatively charged
        and histones positively charged, it proposes that
        histone and DNA binding happens by electrostatic interactions.

        %% We therefore advance the hypothesis that the basic proteins of
        %% cell nuclei are gene inhibitors, each histone or protamin being
        %% capable of suppressing the activities of specific groups of
        %% genes. Such an hypothesis will, we believe, not only serve to
        %% explain the different properties of cell nuclei, but will also
        %% give some indication of the mechanism by which the nucleus
        %% effectively participates in the process of cell differentiation.
        \item[\cite{stedman1951main-histones-separation}]
        Demonstration that histones are not homogeneous and can be
        separated in two parts, main and subsidiary histones,
        which are now known as core histones and histone H1.
        Further demonstration that even these two fracations are not
        homogeneous themselves, that distribution is tissue specific,
        but further fractionation is technically difficult.
        Proposal that histones play a role in gene regulation based
        on the tissue specificity of certain histones.

        %% On this paper they separate F2a into two parts (into what
        %% is now known as H2B and H4).  There are 3 other papers by
        %% the same lab, before this one, where they identify the other
        %% histones but this is the last.  The 3 other papers are
        %% referenced in the introduction of this one.
        \item[\cite{philips-and-johns1965-fractionation}]
        The last on a long series of papers about fractionation and
        identification of histones leading to the identification of
        the five histones types known today.
        This was made possible by the development of chromatographic
        fractionation methods and gel electrophoresis techniques in the
        late 1950s.

        %% They didn't actually refer to 200bp on this paper, there is
        %% no ladder on the figure.  It's Kornberg on the notes of his
        %% references that mentions:
        %%    "the extent of digestion and size of the pieces are
        %%    from D. R. Hewish, personal communication
        %%    (the size was determined by velocity sedimenta-
        %%    tion in alkali and should be regarded as only
        %%    approximate)."
        \item[\cite{hewish1973-200bp-pieces}]
        Digestion of chromatin by nucleases yields \SI{200}{\bp}
        fragments.
        ``It is proposed that chromatin has some simple, basic, repeating
        substructure with a repetitive spacing of sites that are
        potentially accessible to the Ca--Mg endonuclease''.

        \item[\cite{olins1974-nu-bodies}]
        First view of the ``beads on a string'' structure
        \frefp{fig:intro:beads-on-a-string}.
        These results were surprising at the time, and initially
        regarded as artifacts, as it departed from the then current
        view of chromatin structure as a uniform superhelical
        coiling of the DNA itself, stabilised by histone interactions
        \citep{pardon-wilkins-1972model}.

        \item[\cite{anna-isenberg-1974-h2a-h2b}]
        Demonstration of strong association between H2A--H2B.

        \item[\cite{kornberg1974-results}]
        Demonstration that histones form a (H3--H4)\textsubscript{2} tetramer
        an H2A--H2B dimer.

        %% Beautiful paper, with the best abstract ever.  This is it on its
        %% entirety: "Chromatin structure is based on a repeating unit
        %% of eight histone molecules and about 200 DNA base pairs."
        \item[\cite{kornberg1974-model}]
        Basic structure of the nucleosome is proposed as a repeating
        unit of chromatin with eight histones, two of each, and about
        200 base pairs of DNA.

        \item[\cite{bradbury1975-histone-nomenclature}]
        The current histone nomenclature is presented at Ciba foundation
        symposium, and submitted to IUPAC for official standardization.

        \item[\cite{richmond1984-7angstrom}]
        Crystal structure of the nucleosome solved with intermediate
        resolution of \SI{7}{\angstrom}.
        Reveals a disk shaped (H3--H4)\textsubscript{2} symmetric
        tetramer at the center of the nucleosome, an H2A--H2B dimers
        on each face of the disk, and the DNA sequence around it.

        \item[September 1986]
        Author of this thesis is born in Portugal.

        \item[\cite{arents1991-31angstrom}]
        Crystal structure of the nucleosome solved with high
        resolution of \SI{3.1}{\angstrom} showing individual
        histone fold and the ``handshake'' motif.

        \item[\cite{luger1997-28angstrom}]
        Crystal structure of the nucleosome solved with high
        resolution of \SI{2.8}{\angstrom} showing atomic interactions
        between the histones and their internal structure.
      \end{description}

\section{Fluorescence microscopy}

  Fluorescence microscopy is an essential tool in the field of life
  sciences for the visualization of cells and tissues.
  Fluoropores emit light in a specific wavelength after
  being exposed to another specific wavelength, named
  emission and excitation wavelength respectively.
  Illumination of a sample with the excitation wavelength, and filtering
  out all but the emission wavelength from the formed image, allows
  the localization of fluorophores in a dark background.
  If specific molecules and cells can be associated with a fluorophore,
  so can their localization can be infered, and by choosing fluorophores
  with different emission wavelengths multiple molecules can be
  simultaneously identified.
  It would not be possible in normal optical microscopy.

  \subsection{Fluorescent proteins}
    Several fluorescent labels have been developed with different
    properties, each best suited to different techniques.
    These are often split into two major classes: synthetic fluorescent
    molecules such as DAPI or Fluorescein, which either bind to the molecule of
    interest or can be chemically linked to it; and Fluorescent Proteins (FPs)
    which are genetically encodable and can be expressed by the cells themselves.
    FPs allow tagging of individual molecules inside live cell, allowing
    visualization of dynamics in real-time live-cell imaging.
    Synthetic molecules are added exogenously, which often requires cell
    fixation and permeabilization disturbing intracellular structures.
    %% TODO mention synthethic moleculus that bing to proteins with tag like snap
    %%      Over simplistic?  Fusion proteins for luciferase, etc

    The \species{Aequorea} jellyfish Green Fluorescent Protein (GFP) was
    the first FP to be expressed recombinantly.
    Although originally discovered in 1962 \citep{shimomura1962-gfp-discovery},
    it was only in 1994 that it was used as a gene expression
    marker, by being placed under the control of \gene{T7} and
    \gene{mec-7} promoters, in \species{E. coli} and \species{C. elegans}
    respectively \citep{gfp-first-expression-marker}.
    %% when was GFP first used as a gene tag?

    Since then, many GFP variants have been engineered by mutating its
    original nucleotide sequence.  These not only increase fluorescence
    quantum yield, photostability, and improved folding, but also change
    the excitation and emission spectrum, providing a wide range of GFP
    derivatives with different colours.  FPs in other organisms were also
    discovered and were themselves engineered to provide further improved
    variants \citep{FP-color-palette}.

    An important class of FPs that have since been developed is the
    photo-controllable FPs whose
    fluorescent properties can be controlled by excitation with specific
    wavelengths allowing for individual cells and proteins to be optically
    labeled making this proteins specially useful for studies of cell
    lineage and protein movement.
    They are phenotypically split into three categories:

    \begin{description}
      \item[photo-activatable]
        Irreversible dark to bright state conversion.
        PA--GFP (Photo Activatable GFP), the first PA--FP to be
        reported, was developed by mutating the original GFP
        \citep{pagfp-discovery} and is
        still the only green PA--FP.
        It allows to mark and track a subset of selected molecules
        within cells.

      \item[photo-switchable]
        Irreversible conversion from one bright state to another bright
        state with a different emission wavelength.
        Like PA--FPs, they allow the tracking of a subset of selected
        molecules with the added advantages that the whole set is visible
        before the switch, and that the non-switched molecules continue
        to be visible.
        The development of proteins from this group started with Kaede FP
        \footnote{Kaede is named after maple leaf in Japanese but it was
        actually cloned from \species{Trachyphyllia geoffroyi} which is a
        stony coral.} whose wild type form is an obligate homotetrameric
        complex \citep{kaede-discovery}, thus making it
        unsuitable for use as a genetically encoded fusion tag.
        Monomeric Kaede-like FP have since been developed such as
        mEos2 \citep{meos2-discovery}.

      \item[reversible photo-switchable]
        Reversible conversion between dark and bright states, \eg Dronpa.
        These FPs have been mostly used in single molecule localization
        microscopy, a type of super-resolution microscopy, due to the
        low energy required for the transition between states.
        %% We could mention RESOLFT but that would be too much detail.

    \end{description}


  \subsection{FRAP}

    Fluorescence Recovery After Photobleaching (FRAP) is an optical
    microscopy technique to measure the dynamics of fluorescently
    tagged molecules.
    Originally developed in the 1970's for quantitative dynamics of lipids
    in cell membrane under the name Fluorescence Photobleaching
    Recovery (FPR) \citep{axelrod1976mobility}, it has
    been extensively used to obtain qualitative and quantitative
    insight on the kinetic properties of proteins since the development
    of GFP tagging.
    From studies of X, Y, and Z, to I, J, and K, FRAP is an ubiquitous
    technique in the field of cell biology with continual reviews being
    published\addref{for all the reviews about it every year}.

    In this technique, fluorescently tagged molecules in a small region
    are photobleached.  The fluorescent intensity in the region is
    measured over time to obtain a recovery curve.  The recovery is a
    function of the molecule dynamics, it represents the unbleached
    molecules, from outside the bleached region, moving into the
    photobleached region.  The molecule dynamics is determined by the
    transport and diffusion rates, as well as binding interactions.

    In the simplest case, a FRAP recovery curve is analysed
    qualitatively.  By looking at the plotted data, one assesses
    whether recovery is fast or slow, presence of an immobile
    fraction and binding interactions, and how this changes
    between different cases \fref{fig:intro:frap-curve-example}.

    \begin{figure}
      \centering
      \missingfigure{shwo recovery curve example, with parts.}
      \caption{}
      \label{fig:intro:frap-curve-example}
    \end{figure}

    In quantitate FRAP, data analysis involves fitting the recovery
    data to an idealized mathematical model.
    Many different models exist with different assumptions for the model
    \fref{fig:intro:frap-model-components}, and dependent on different
    parameters.
    Kinetic parameters are typically \Kon{} and \Koff{}, for
    binding and unbinding rate constants, and $D_{f}$ for diffusion constant.
    Different parameter values for the model are tested, until a curve is
    obtained that fits best to the measured recovery.

    \begin{figure}
      \centering
      \missingfigure{table of different assumptions as in Macnally's 2010 review}
      \caption{blah blah blah, there's so many models, how can anyone trust
               any of this?}
      \label{fig:intro:frap-model-components}
    \end{figure}

    Two factors contribute for the widespread use of this technique:
    it doesn't require specialized instruments and can be performed with
    any standard confocal microscope or widefield microscope with a laser
    module; and recovery curves are easy to understand making it easy to
    identify artifacts.  This is not the case with alternative techniques
    such as FCS (Fluorescence Correlation Microscopy) or single molecule
    tracking.

    The technique however has implicit assumptions which are applicable
    for most biological situations:

    \begin{itemize}
      \item The biological system has reached equilibrium and the equilibrium
        is maintained throughout the experiment, \ie{} the kinetic parameters
        must remain constant throughout the experiment;

      \item distribution of tagged protein mimics the endogenous protein;

      \item the binding sites are part of a large, relatively immobile
        complex, on the time and length scale of the recovery.
    \end{itemize}

    These assumptions become difficult to maintain as the experiment
    times increases, a requirement for slow moving molecules.  This
    is discussed again in \Cref{ch:kill-frap}.

  \subsection{iFRAP}


  \subsection{FLAP}
