\chapter{FRAP with histones}
\label{ch:frap}

%% chapter concept: in this chapter goes the whole kill FRAP project. I started
%% positive that this should work and so should the text. We start by studying
%% the technique and list the assumptions it requires. We do one experiment and
%% face some problems. Each problem has its own section that ends in a solution.
%% When the solution was not done, we could offer one. The last one is the cause
%% that this is not possible and has no solution. Conclusion lists all problems
%% and their solutions again.

%% FIXME 2nd year report
%%My overall objective is to measure \textit{in vivo} the role of different nucleosome regions
%%in chromatin dynamics. Based upon the workflow established in my first year, I aim to obtain
%%the mathematical analysis knowledge that would enable examination of FRAP data and obtain
%%accurate quantitative data. This defines a series of objectives:
%%
%%\begin{enumerate}
%%  \item Create stable cell lines expressing a FP-tagged Sin$^{-}$ mutant;
%%  \item Track cells moving during long time FRAP experiments;
%%  \item Obtain recovery curves from FRAP experiments;
%%  \item Measure kinetic properties from the recovery curves.
%%\end{enumerate}

The building block of eukaryotic chromatin structure is the nucleosome, comprising
\SI{147}{\bp} of DNA wrapped around an octamer of two copies of core histones H2A,
H2B, H3 and H4. They are arranged in a linear chain separated by DNA linkers, and
can be further compacted into higher order chromatin structures. But chromatin extends
well beyond DNA compaction, it is a dynamic complex that controls access to genetic
information by undergoing reconfiguration of its structure.

Since nucleosomes are chromatin's basic structural unit, it is the interactions
within these structures that determine its configuration. As such, these local
reconfigurations can be accomplished by changing nucleosome structure or altering
its composition by post-translational modifications or incorporation of variant
histones.

These changes, and through them access to genetic information, are usually performed
by protein complexes. Mutations of the histones can relieve the need of these complexes
by affecting the stability of the nucleosomes so it is possible to gain insight on
the mechanism of these complexes by studying the kinetic alterations these mutants create.

Studies of this mutations have all been done \textit{in vivo} or yeast. We are trying
to validate to measure their effects in the context of chromatin by using live cells.

\section{The FRAP model}

  Fluorescence recovery after photobleaching (FRAP) is an optical technique
  that reveals the dynamics of fluorescently tagged molecules within live cells.
  The tagged molecules inside a small region are irreversibly photobleached by
  action of a high-power focused laser beam and the recovery rate of fluorescence
  is measured. The recovery rate is interpreted as unbleached molecules,
  which were outside of the region at the time of photobleaching, moving into
  the bleached area, replacing the bleached molecules. It is assumed that the
  the fluorescence recovery reflects the protein natural movement.

  This technique has been extensively used to obtain qualitative and quantitative
  insight on the kinetic properties of proteins. Development of this technique has
  led to complex models that are both more precise and accurate than simple models
  based on inverse exponential decay. These take into account important parameters
  that were previously discarded but have since been shown as important, such
  as diffusion and the bleach spot profile shape.

  However, despite their sophistication, these models of recovery make certain important
  assumptions that are hard to maintain for long experimental observation times:

  \begin{itemize}
    \item the biological system has reached equilibrium before photobleaching;
    \item total amount of both Fluorescent Protein (FP) fusion protein and its
          binding sites remain constant over the time course of the recovery;
    %% FIXME we should probably group the 2 before into one
    \item distribution of tagged molecule mimics the endogenous protein;
    \item the binding sites are part of a large, relatively immobile complex, at
          least on the time and length scale of the recovery.
  \end{itemize}

  FRAP has been successfully used before to show that different core histones have
  different kinetics and populations. We expected to be able to use the same approach
  for \textit{in vivo} validation of previously identified differences between
  wild type and mutant histones.

  %% plot function for long recovery times to see minimum analysis time to see the difference
  %%it's usually necessary to see the whole recovery but for these that's just not possible
  %% XXX nah! This already points for failure so we will place the fig later on

\section{Cell movement}

  The first problem one faces when dealing with long time imaging, is movement of the cells.
  Initial experiments were done with HeLa cells over a time course of 8 hours. We explored
  a couple of approaches
  
  %% try with higher density cells (lowered but did not solve)
  %% primary cells with contact inhibition (didn't solve and is very hard to deal with plus cell line is not standard)
  %% computational approach solved the problem (cropreg + stackreg) together with higher cell density

  Using the program CropReg developed during my second year I managed to
  track cells moving during the long time experiments. However, this approach
  did not account for rotational movement of the nucleus on the Z axis which
  is now solved by using the ImageJ plugin StackReg.

  We expected to solve this problem in a non-computational way when using
  cells with contact inhibition but this did not happen.
  \missingfigure{horse cells moving around}

  %% That's the first problem, the cell is moving around. It's not a show killer, just more difficult.
  %% Fixes for these include simple scripts, ImageJ plugins. Some microscope already have software built
  %% in to track cells and keep them on focus. It didn't work for us because if a cell on the field view
  %% was in mitosis, it would mess up the algorithm. However, when the cell moves, the nucleus reshapes
  %% itself, it's not just dragged which makes

    %% \missingfigure{cell moving an reshaping the nucleus as it moves}

  %% A fix for this could have been use of primary cell lines that have contact inhibition (but they are
  %% a pain to transfect). Must try transfection by electroporation
  %% we did use horse cells in the end
    %% \missingfigure{table with common used cell lines, human and mouse, that still show contact inhibition}

\section{Cell cycle}
  %% If cell enters S phase, binding sites double but system must stay in equilibrium.

  %% Cell stays on G1 for longer than G2 so it's prefered.

  %% What's the maximum time that a cell stays on G1? I searched for cells in mitosis,
  %% tracked them for a few hours until they were in G1 and FRAP both of daughter cells.
  %% However, even on the first hours of G1, Kimura found fancy dynamics for some DNA binding proteins.
  
  %% Drugs should be avoided. Different drugs had different effects on Kimura and Cook paper.

  %% contact inhibition could also fix this. But would it still fix the problem of chromatin movement

\section{Expression levels}

     %% on the case of proteins with special control of expression, may make sense to look into it.
     %% Specially if the proteins will have very slow exchange rates this may affect its distribution.

%%      \missingfigure{qPCR showing expression of tagged and endogenous histones on different cell phases
%%      \missingfigure{western showing comparison of endogenous vs tagged
%%      \missingfigure{FRAP recovery, almost 20\% is free pool
%%      \missingfigure{western of chromatin bound vs unbound fractions. Subcellular fractionation

      %% Solution may be to clone to whole gene. In some cell lines (DT40) it's very easy to knock in
      %% a gene and tag the endogenous. When zinc-finger techonology becomes more accessible it may
      %% become possible to use it on more common human models.

    \subsection{protein distribution}
      %% Kimura and Cook mentioned on their original paper that the distribution of the tagged H3 was
      %% different from the endogenous and showed it by mixing distribution of DNA with H3--GFP. This
      %% shuld not be a problem if expression levels are the same.
      %%\missingfigure{similar image as Kimura and Cook (mix channels) + Plot of relative intensities of each pixel}


\section{Chromatin movement}

    %% The chromatin moves. It's not possible to correct for this movement on the model unless it's possible to see
    %% it. However, histones are usually the ones used to see it. This seems to be the limit.

%    \missingfigure{cells with H3-PA-GFP. Activate a few circles and see distribution 1h later}
%    \missingfigure{see effect of different drugs on this movement}

    %% not homogeneous. Different parts of the chromatin should have different recoveries.

\section{Discussion: why histone dynamics is not measurable}

    %% FRAP is not goof for very long time frames. Reference papers where it was used incorrectly?
    %% see Tim's paper

\section{Conclusions}
  We have show that this is not possible using FRAP technique.
  
  Another group has recently
  published some work where they claim doing single molecule tracking of H4 tagged with PAGFP
  and measure movement over \SI{90}{\ms}\addref{Saera Hihara et al 2012}. They kind of claim
  a small difference on movement between interphase chromatin and mitotic chromosomes but
  I'm not convinced. Plus, the difference should be the highest between these two so even if
  the difference they see is real, it's unlikely to be useful in seeing a difference between
  WT and our mutant histones. To control against the chromatin movement they claim that the
  centroid of the nucleus does not changes but over that time frame it should not. It's the
  insides that count and that wouldn't change it.
  


