\section{Discussion}

    We wished to quantitatively determine the
    effect on human chromatin dynamics
    of SIN mutations in core histones H3 and H4 known
    to be destabilising \textit{in vitro} and to affect
    cell growth in \species{S. cerevisiae}.
    We set out to use a previously reported circle FRAP model
    which accounts for multiple factors in a typical
    FRAP modelling \citep{mcnally-frap-code}.

    However, FRAP recovery is incomplete even
    after 8~hours for core histones \citep{KimuraCook}.
    This led us to address a series of technical challenges in
    collecting valid quantitative recovery data over extended time periods.

% Cell movement

    The first problem faced was cell motility, which
    is an expected property of actively dividing cells.
    We attempted to reduce motility by taking advantage of
    the fact that many primary cells display contact inhibition of
    locomotion and proliferation when they reach high densities.
    This contact inhibition is a natural mechanism
    that controls cellular growth in
    multicellular organisms, and results in a stop in proliferation
    with the formation of a monolayer of healthy cells in tissue culture.

    However, the approach has disadvantages including
    increased cell handling and reduced transfection efficiency.
    The potential inability to compare results with published data for
    immortalised cell lines such as HeLa is also undesirable.

    Despite achieving a monolayer of healthy cells that
    could be maintained stably over 2 weeks,
    individual transfected primary horse fibroblasts still showed motility
    despite exhibiting overall characteristics of contact inhibition.
    Furthermore, nuclei in these cells displayed a helical motion
    on the direction of cell movement \frefp{fig:kill-frap:confluent-horse}.

    The possibility of chemically inhibiting
    cells to reduce motion was considered
    since previous FRAP experiments with core
    histones were performed using multiple inhibitors
    of protein synthesis \citep{KimuraCook}. However, these studies revealed
    inhibitor-dependent variations in kinetics and the authors qualified
    their conclusions about the absolute accuracy
    of the histone exchange parameters measured.

    To better address the problem of cell motility we
    instead developed a computational approach
    by writing the program CropReg for cell tracking by normalised
    cross-correlation template matching.
    Using automated analysis enabled us to process
    the large numbers of cell images
    required to provide statistically valid
    quantitative measurements of core histone exchange.

% Compositional changes

    The second challenge to measuring core histone exchange by FRAP is that
    a chemical equilibrium is required between freely diffusing proteins and
    formation of a complex. Although absolute equilibrium is unlikely
    in the dynamic cell environment undergoing
    complex transcriptional and translation responses anyway,
    DNA replication involving polymerase passage
    and repackaging of the duplicated genome
    in S~phase will certainly unbalance any equilibrium.

    Chromosome compaction in mitosis also generates a chromatin environment
    that is distinct from interphase.
    This limits FRAP experiment to either \G1{} or \G2{} phases.
    The HeLa cell cycle has a typical \G1{} phase of 11.7~hours
    and a \G2{} phase of 3~hours \citep{HeLaCellCycle}
    so the extended time periods needed for FRAP of core histones requires
    starting FRAP early in \G1{} \frefp{fig:kill-frap:cell-cycle}.

    Post-mitotic chromosomes take approximately 2~hours
    to migrate within the nucleus
    and rebuild the interphase nuclear architecture during early \G1{}
    \citep{visualizationG1chromosomes,earlyg1position,RelativeChromosomePosition}.
    This defines the window for extended FRAP experiments
    from approximately 3 to 11 hours after mitosis
    in HeLa cells, although cells lines with even longer
    \G1{} phase could also be used \citep{PancreaticCells}.

    We wished to avoid the use of drugs for cell
    cycle arrest since this has been
    shown to influence FRAP results \citep{KimuraCook}.
    We also discounted serum starvation to move cells into the
    quiescent \G0{} phase since this could affect
    the relevance of measuring core histone
    exchange \citep{SerumStarvation}.

      \begin{figure}
        \centering
        %% based on original code from Robert Vollmert
        %% http://www.texample.net/tikz/examples/pie-chart/
        \newcommand{\slice}[4]{
          \pgfmathparse{0.5*#1+0.5*#2}
          \let\midangle\pgfmathresult

          % slice
          \draw[thick,fill=black!10] (0,0) -- (#1:1) arc (#1:#2:1) -- cycle;

          % outer label
          \node[label=\midangle:#4] at (\midangle:1) {};

          % inner label
          \pgfmathparse{min((#2-#1-10)/110*(-0.3),0)}
          \let\temp\pgfmathresult
          \pgfmathparse{max(\temp,-0.5) + 0.8}
          \let\innerpos\pgfmathresult
          \node at (\midangle:\innerpos) {#3};
        }
        \begin{tikzpicture}[scale=3]
          \newcounter{a}
          \newcounter{b}
          %% Total cell cycle is 24.5 hours, G1 is 11.7h, S is 8.8h,
          %% G2 is 3h, M is 1h. The problem is that the counters can't handle
          %% decimal places so we have a variable with the actual time for
          %% the text, and another one times 10 to calculate the angle.
          \foreach \p/\t/\l in {117/11.7/\G1, 9/0.9/M,
                                31/3.1/\G2, 88/8.8/S}
            {
              \setcounter{a}{\value{b}}
              \addtocounter{b}{\p}
              \slice{36*\thea/24.5} % we multiply by 36 instead of 360 because
                    {36*\theb/24.5} % the time is already times 10
                    {\l}{\t{} hours}
            }
        \end{tikzpicture}
        \captionIntro{HeLa cell cycle phases and timing}
          {
            Under optimal growth conditions the HeLa cell has a median
            doubling time of 24~hours, with \G1~and
            S~phases of 11.7 and 8.8 hours respectively \citep{HeLaCellCycle}.
          }
        \label{fig:kill-frap:cell-cycle}
      \end{figure}

    Instead, we developed a procedure to track
    progression of cells manually during mitosis
    where visual identification of the cell cycle is possible.
    This allowed us to minimise the variations of normal cell growth
    and to identify individual cells exactly 3 hours after start of \G1{}.
    This has the added advantage of allowing time for maturation of GFP
    expressed during the establishment of interphase.
    The time interval between images during manual selection was increased and
    both resolution and laser intensity were reduced
    to minimise phototoxicity or bleaching.
    This resulted in a set of selected early \G1{}
    cells suitable for FRAP experiments.

    The fluorescently tagged histone proteins are constitutively expressed
    under the control of an EF-1\textalpha{} promoter,
    so they lack the 3' regulatory features of native histone genes.
    This regulation does not follow the normal
    expression program of a histone gene
    and could affect the distribution of the histone in chromatin.
    Constant expression of tagged histones by a strong constitutive promoter
    will enrich them in the \G1{} and early S~phase pools
    making subsequent incorporation in euchromatin more likely,
    relative to mid-late S~phase where heterochromatic sequences
    are replicated and packaged \citep{DNA-replication-timing}.

    A more realistic tagged histone expression profile could be achieved using
    flanking regulatory regions from native histone genes,
    as demonstrated for H3 and CENP--A \citep{pMH3-plasmid,Kevin-pCA-TAG}.
    Another potential solution is to insert GFP
    in-frame into the native gene locus by genome engineering,
    although the redundancy between the multiple canonical histone genes means
    that identifying the most appropriate isoform to target
    could introduce complexities.

    Protein synthesis inhibitors were used by \citet{KimuraCook}
    to address this issue,
    but this has the disadvantage of potentially affecting
    many other processes as discussed above.

      %% TODO: would be cool to create this figure
%      \begin{figure}
%        \centering
%        \missingfigure{a schematic of cell cycle, soluble pool}
%        \captionIntro{Distribution of tagged and endogenous histones during cell cycle}
%                     {
%                       This would be at least 3 different subplots. The first
%                       and the second are like the ones in Fig 7A of Kimura and
%                       Cook paper. The third one would show the ratio of each
%                       histone over time, i.e., 100\% tagged during all cell
%                       cell cycle and some endogenous during S phase. In this
%                       plots, also note where euchromatin and heterochromatin
%                       are replicated.
%                     }.
%        \label{fig:kill-frap:messy-histone-expression}
%      \end{figure}

% Movement of the reference

    The final challenge to measuring core histone
    exchange by FRAP that we identified
    was non-homogenous regional movement of chromatin itself.

    One possible cause for this movement is chromatin repackaging
    during DNA replication which we addressed by selecting cells at
    early \G1{} phase.
    One other cause is chromatin remodelling as part of a DNA damage
    response caused by the FRAP photobleaching event itself.
    The phototoxicity effects of a FRAP experiment are often dismissed
    on the basis that the photobleaching event of a typical FRAP
    experiment does not affect cell viability
    \citep{kruhlak2000reduced, KimuraCook, carrero2003using} but the
    DNA damage that such experiment may introduce, and the effect that
    the repair response of such damage may have on chromatin
    reconfiguration, has not yet been addressed.
    Still, the use of laser beams of different wavelength and longer
    exposure times are often used to introduce both single and double
    DNA strand breaks on which FRAP experiments are then performed
    \citep{stixova2014advanced, mari2006dynamic, kim2002specific}.

    Independently of the cause for the chromatin movement,
    it undermines the assumption of FRAP analysis that binding sites
    remain immobile throughout the FRAP experiment.
    This assumption is required to interpret recovery
    as the rate of movement of freely diffusing unbleached molecules into the
    bleached area which allows the kinetic
    rates \kon{} and \koff{} to be estimated.
    If chromatin binding sites also move then the recovery curve becomes a
    much more complex function of both binding site movement and free diffusion.

    Chromatin movement is recognisable
    by changes in the intra-nuclear features of the fluorescent chromatin
    and by changes in the circular bleach spot.
    Although some of these effects are subtle when observed by photobleaching,
    the photoactivation of an equal circular spot demonstrates
    clear non-homogenous reshaping of chromatin.
    Equivalent chromatin movement has also been reported
    for H4--PAGFP in strip photoactivation \cite{H4PAGFP-chromatin-movement}.

    The movements we observed were in the
    range of \SI{4}{\um}, which is double the size of the bleach spot,
    and exhibited complicated shapes reminiscent of channelling.
    This is consistent with chromosome distribution in nuclei that is
    territorial on the scale of \SI{5}{\um} \citep{sun2000size}
    separated by interchromosomal channels of
    \SIrange{10}{100}{\nm} \citep{gorisch2005histone}.

    The clarity of H2B--GFP imaging by photoactivation
    suggests the opportunity to analyse the
    paths taken by diffusing core histones.
    For example, simultaneous use of combined
    photoactivation and photobleaching
    of complementary dimer and tetramer histones could
    enable relative diffusion rates and paths to be determined.
    Alternatively, an enzymatic mechanism to incorporate a
    complementary photo-differented label
    into DNA  would facilitate masking for
    the original location at the same time as tracking the histone diffusion
    and enable quantitative FRAP.
    Nevertheless, it is important to recognise
    that such experiments would test the
    resolution and sensitivity of microscopes.

\section{Conclusion}

    Since its inception over 30 years ago, FRAP has been continuously
    improved
    through technical capabilities of light microscopy
    and sensitive kinetic models that are now able to take into account
    an increasing number of biophysical features such as container size,
    non-homogeneous distribution of fluorescence, and profile of bleach spot.

    Despite these advances, the ability to perform FRAP
    over extended time periods of several hours for highly stable complexes
    such as core histones is limited by the dynamic nature of the cell.

    We overcame the challenges of cell motility
    and selection of cells in \G1{} phase,
    but were not able to develop a method to adjust
    for changes in chromatin structure within the cell nucleus.
    While a photobleached spot appears stable and
    can be tracked over several hours,
    small natural disturbances and non-homogeneous diffusion
    impact on photorecovery
    and estimation of kinetic parameters.
    We find that FRAP is suitable for semi-quantitative
    estimates of slowly diffusing molecules
    but not for the precise quantitative comparisons
    required to compare core histone mutations.

    Ultimately, the issue of long observation times stems from the
    requirements of FRAP models to achieve full recovery of the mobile
    populations being measured.  Alternative techniques capable to
    detect the motion of isolated molecules would reduce the
    requirement for observation times and overcome the issue of
    chromatin movement.  However, the high concentration of histones
    in the nucleus makes it challenging to observe individual
    molecules as required for approaches such as single particle
    tracking.

    Fluorescence Correlation Spectroscopy (FCS) is a fluorescent
    microscopy technique providing estimates of dynamic parameters by
    using fluorescence fluctuations in a femtolitre volume.  A
    variation of FCS named Raster Scan Image Correlation Spectroscopy
    (RICS) has recently been developed that uses larger image regions
    such as an entire cell nuclei, and accounts for the spatial
    variation of dynamics \citep{digman2005rics}.  This technique
    could potentially provide not only a method to compare core
    histone mutations \textit{in vivo} but also their effect in
    different chromatin domains.

    Overall, the current period of rapid technological advances in
    cell biology research means that new techniques such as RICS offer
    hope to address specific challenges such as histone mobility for
    which existing approaches such as FRAP are poorly suited.

%  Single-molecule imaging of histones for short period of times in
%  live cells has recently been reported using super-resolution
%  imaging\addref[nature methods 7(9):717-719, 2010 and nature methods
%  8(1):7-9, 2011].

%  Also, use of PAGFP has been used to measure dynamics of H4 over
%  \SI{90}{\ms} reporting differences between interphase chromatin and
%  mitotic chromosomes\addref[Saera Hihara et al 2012].  However, the
%  difference between these two phases is the highest and might not be
%  comparable to the difference between histones variants\todo{study
%  this. Someone must have measured this}.
