\section{Introduction}

  \subsection{Histone contribution to nucleosome dynamics}

    The building block of eukaryotic chromatin structure is the nucleosome, comprising
    \SI{147}{\bp} of DNA wrapped around an octamer of two copies of core histones H2A,
    H2B, H3, and H4 \citep{luger1997crystal}.
    Nucleosome core particles are arranged in a linear chain separated by DNA linkers, and
    can be further compacted into higher order chromatin structures.
    The role of chromatin not only achieves DNA compaction,
    but also provides a dynamic complex to mediate access to genetic
    information through its capability to undergo reconfiguration of
    structure \citep{flaus2017unlocking}.

    Local remodelling of chromatin can be achieved
    by changing nucleosome structure or altering its composition.
    This can be performed intrinsically through post-translational modification
    or incorporation of histone variants,
    causing nucleosomes to alter DNA sequence preferences or recruit other proteins.
    Alternatively, chaperones and ATP-dependent chromatin remodelling complexes
    can act extrinsically to alter nucleosome structure or position.

  \subsection{Histone SIN mutants}

    The archetype of ATP-dependent nucleosome remodelling is the SWI/SNF complex
    whose deficiency causes growth defects in yeast.
    This complex was identified independently through screens for
    mating type SWItching \citep{SWI-mutants}
    or Sucrose Non Fermentation \citep{SNF-mutants-original-discovery, SNF-mutants2}.

    A set of mutations were identified that compensate
    for the loss of the SWI/SNF complex
    that are collectively known as SIN mutations because they
    provide SWI/SNF INdependence \citep{kruger1995amino}.
    A subset of these SIN mutations result in single amino-acid changes to core histones,
    providing a direct link between SWI/SNF and chromatin and suggesting
    that the mutated histone protein residues influence the same 
    nucleosome dynamic pathways leveraged by the enzyme and are therefore
    of major importance in the nucleosome structure.
    This predicted that the stability of SIN mutant containing nucleosomes would be affected in chromatin.
    The hypothesis has been tested \textit{in vitro}
    where it was observed that SIN mutant nucleosomes display higher 
    thermally driven nucleosome sliding mobility \citep{flaus2004sin}
    and that the mutated residues alter histone-DNA contacts in
    crystal structures \citep{muthurajan2004crystal}.

    However, the effect of histone protein SIN mutants
    in the more complex \textit{in vivo} chromatin environment 
    of mammailian cells has not been demonstrated. This is important to
    validate the functional significance of the residues
    and their implications for the basis of high conservation of 
    histone protein sequences in eukaryotes.

  \subsection{Fluorescence Recovery After Photobleaching}

    Fluorescence Recovery After Photobleaching (FRAP) is an optical technique
    that reveals the dynamics of fluorescently tagged molecules within live cells.
    Tagged molecules inside a small region are irreversibly photobleached by
    a high power focused laser beam and the recovery rate of fluorescence
    in the bleached area is measured. The recovery rate is interpreted as unbleached molecules
    from outside of the region at the time of photobleaching diffusing into the bleached area.
    It is assumed that this fluorescence recovery reflects natural protein movement.

    FRAP can be described by a simple chemical equilibrium:

    \begin{displaymath}
      F + S \overset{k_{on}}{\underset{k_{off}}{\rightleftharpoons}} FS
    \end{displaymath}

    where $F$ represents freely diffusing proteins, $S$ represents immobile vacant
    binding sites, and $FS$ the complex between the two, when the protein is bound
    to the binding site. The value of \Kon{} and \Koff{},
    are estimated from the rate at which photobleached $F$ is replaced in the $FS$ complex.

    Ongoing development of FRAP has led to increasingly complex models
    that are both more precise and accurate than simple
    inverse exponential decay \citep{mcnally-frap-2010}.
    Despite their sophistication, these models require assumptions
    that are difficult to maintain over long experimental observation times.
    Firstly, equilibrium must be maintained throughout the entire experiment 
    so that both \Kon{} and \Koff{} remain constant.
    This also requires that concentrations of both $F$ and $S$ remain constant.
    Secondly, distribution of the fluorescently tagged molecule must mimic the endogenous protein.
    And finally, the binding sites must be part of a large, relatively immobile complex
    on the time and length scale of the recovery.

  \subsection{FRAP measurements of histones}

    FRAP has been extensively used to obtain qualitative and
    quantitative insight on the kinetic properties of chromatin bound
    proteins \citep{phair2000high, essers2005nuclear, agresti2005gr}.
    These sit on the established assumption that chromatin is
    relatively immobile in the interphase nuclei
    \citep{abney1997chromatin} as they show high-mobility and recovery
    happens in the scale of seconds or minutes.  While chromatin
    labelling was originally performed in the DNA with
    dihydroethidium, a derivative of ethidium bromide, H2B--GFP became
    the standard reference as immobile reference in FRAP experiments
    \citep{dey2000bromodomain} once GFP labelled histones became
    available \citep{KevinH2BGFP}.

    %% Histones are often used as the reference for immobile in FRAP.
    %% Of special interest is Dey et al 2000 ``A Bromodomain Protein,
    %% MCAP, Associates with Mitotic Chromosomes and Affects G2-to-M
    %% Transition'', which Kimura and Cook use as an example of FRAP
    %% done on histones, but actually, they used H2B-GFP as an example
    %% that showed no FRAP recovery after 100 minutes.  From Dey et
    %% all:
    %%
    %%     "As shown in Fig. 6B, GFP-MCAP fluorescence was reduced to
    %%      background levels immediately after photobleaching but
    %%      recovered ~86% of its intensity within 1 min. By contrast,
    %%      histone H2B-GFP did not recover any fluorescence over this
    %%      period. These results indicate that while histone H2B, a
    %%      stable component of chromatin is immobile, [... talk about
    %%      MCAP]"

    However, dynamics of histones were still studied by FRAP
    \citep{KimuraCook}.  This study found that there were multiple
    populations of H2B, each with their own exchange rate.  A small
    proportion of about \pcent{3} had a rapid recovery within minutes,
    a larger proportion of \pcent{40} had a slow recovery with a
    \halflife[$\approx$\SI{130}{\min}], while the rest of \pcent{53}
    had a very slow recovery with a \halflife[$>$\SI{510}{\min}].  The
    slower moving H3 and H4 had no rapid exchanging fraction, only
    slow and very slow populations, \pcent{16} and \pcent{68}
    respectively, with the same \halflife[] as the H2B populations.
    In addition, the authors identified a mobile fraction of freely
    diffusing histones of \pcent{4} for H2B, and \pcent{16} for H3 and
    H4, the high value for H3 being explained as unregulated
    expression of the GFP tagged H3 and so, an artifact of the
    experiment.
    %% This mobile fraction is odd.  They claim it is freely moving
    %% histone but then claim that it was the recovery before the
    %% first post-bleach image could be acquired.  To me that sounds
    %% like a non-bleached fraction.

  \subsection{Aims}

    In order to observe the implications for nucleosome structure and function for
    the histone SIN mutant H4~R45H that exhibited the
    highest increase in nucleosome mobility \textit{in vitro}
    we attempted to determine its exchange characteristics by FRAP.
    To achieve quantitative measurements we addressed the multiple technical challenges 
    of measuring subtle kinetic alterations of the nucleosome dynamics 
    over long time periods in live cells. This allowed us to define 
    the limitations of FRAP in observing molecules with extremely slow exchange rates.
