\section{Results}

  \subsection{FRAP analysis in Octave}

    In order to estimate binding constants from FRAP data, we ported
    \textsc{Matlab} code for circle FRAP \citep{mueller2008evidence} to GNU Octave.

    Functions for nonlinear regression were required for the model
    fitting and were replaced by \texttt{leasqr}
    from Octave-Forge's optim package. This function performs the
    Levenberg--Marquardt nonlinear least squares algorithm which is the
    same as the documented method for \texttt{nlinfit}.
    Several sample images were analyzed
    by the Octave port and the fitted values were found to be equal to
    the values reported by the authors using the original code (data not shown).
    Functions for graphical user interaction were mainly required
    for manual selection of Regions Of Interest (ROIs) and options.
    Selection of ROIs was replaced by automatic identifications
    and parameter input was made command-line based for batch processing.

    \subsubsection{ROI selection}

      Bleach spot, nucleus region, and background region ROIs are required
      for the circle FRAP model. The bleach spot measures intensity recovery
      and also models the photobleach profile since it takes into account
      a non-uniform spatial distribution of the bleached spot. The nucleus
      region defines the finite sized nucleus and takes into account
      the fluorescence loss during the photobleaching,
      while a small region outside the nucleus is used for background correction.

      Cells expressing GFP tagged histone proteins have well defined nuclei.
      The large amount of histone protein and specificity for nuclear chromatin
      produces a strong and clear signal for the nuclei contrasted on a dark background (\fref{fig:kill-frap:roi}).

      \begin{figure}
        \centering
        \subbottom[pre-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/roi-prebleach.png}
          \label{fig:kill-frap:prebleach}
        }
        \hfill
        \subbottom[post-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/roi-postbleach.png}
          \label{fig:kill-frap:postbleach}
        }
        \subbottom[pre-bleach $-$ post-bleach]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/roi-subtracted.png}
          \label{fig:kill-frap:subtracted}
        }
        \hfill
        \subbottom[Identified ROIs]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/roi-selected.png}
          \label{fig:kill-frap:selected}
        }
        \captionIntro{Automatic selection of ROIs for FRAP}
          {
            HeLa stable cell line expressing the H4~R45H mutant tagged with YFP,
            are imaged every \SI{30}{\ms} in a confocal microscope. A circular
            shape is used for photobleaching after 100~frames.
            \subcaptionref{fig:kill-frap:prebleach} averaging of 50~pre-bleach
            images removes most of the noise, allowing for a better refined
            ROI;
            \subcaptionref{fig:kill-frap:postbleach} average of
            5 post-bleach images;
            \subcaptionref{fig:kill-frap:subtracted} subtraction of the
            post-bleach to the pre-bleach image, gives a clear indication
            of the bleach spot, as well as faint signal for the nuclear
            region due to unintentional photobleaching;
            \subcaptionref{fig:kill-frap:prebleach} perimeter of the
            automatically identified ROIs superimposed on the pre-bleach
            image: cell nuclei, bleach spot, and background region.
          }
        \label{fig:kill-frap:roi}
      \end{figure}

      Subtraction of the post-bleach (\fref{fig:kill-frap:postbleach}) from
      the pre-bleach (\fref{fig:kill-frap:prebleach}) images displays
      a clear circular shape corresponding to the bleach spot (\fref{fig:kill-frap:subtracted}).
      The centre for this spot was identified by finding the maximum of the convolution matrix
      between the subtracted image and a disk kernel.
      To reduce signal-to-noise ratios, multiple pre and post-bleach images of the cell
      were averaged before image operations.
      While the bleach spot is the most visible feature, there is a faint nuclear region signal
      caused by background photobleaching during image collection.

      An approximate border for cell nuclei could be readily defined using Otsu's method for image threshold.
      As with the identification of bleach spots, multiple pre-bleach images
      were averaged before threshold definition for a more accurate cell border (\fref{fig:kill-frap:selected}).
      Since multiple nuclei often appeared in a single image, the coordinates for the bleach spot
      were used as reference to select the correct nucleus.
      The background region was identified by finding the minimum of the convolution matrix
      between the averaged pre-bleach images and a small square of black intensity values.

    \subsubsection{Batch processing}

      \begin{sidewaysfigure}
        \includegraphics[width=\textwidth]{kill-frap/results/frapinator.png}
        \captionIntro{Frapinator visual log files for batch processing}
          {
            Each FRAP experiment generates a log file with 6 different plots
            displaying the analysed values and the fitting to different models.
            In conjunction with the images in \fref{fig:kill-frap:roi} this
            provides a quick overview of the entire analysis process.
            The top left plot displays the raw intensity
            for the background, bleach spot, and nucleus intensity over the
            duration of the FRAP experiment. This is followed by the normalized
            intensity for the bleach spot which is actually used for the
            fitting. The top right displays the intensity
            profile for the bleach spot, and its fit to a radial profile
            model. The three bottom panels display the data fitted to three
            different models: pure diffusion which has no terms for binding
            constants; full model with a fixed diffusion rate; and full model
            with all the 3 terms.
          }
        \label{fig:kill-frap:frapinator}
      \end{sidewaysfigure}

      The automated identification of ROIs was turned into a stand-alone program, frapinator.
      This has the advanatge that the user can set command-line options without scripting.
      Representative images are saved with results and all intermediate analysis is logged for post-processing analysis.
      These include the automatically identified ROIs and
      plots of measured intensities, bleach spot profile, and fitting to different models \fref{fig:kill-frap:frapinator}.
      Frapinator is available for public download under the GPL \footnote{\url{https://github.com/af-lab/frapinator}}.

  \subsection{Tracking of cell nuclei}

    %% We could show this but it would only be to "encher chouricos"
%    \begin{figure}
%      \centering
%      \missingfigure{Our first FRAP experiment}
%      \captionIntro{Long time series of HeLa cells expressing H2B--EGFP.}
%                   {Cells were transfected with pBOS--H2B-EGFP and imaged for 8
%                    hours, with intervals of 20 minutes.}
%      \label{fig:kill-frap:cell-movement}
%    \end{figure}

    Histone proteins exhibit extremely slow kinetics of exchange
    requiring FRAP to be performed over several hours during which cell movement is
    a particular issue. % \fref{fig:kill-frap:cell-movement}.

    %% This needs some sentences describing the nature of effects of the movement,
    %% and a semi-emirical statement of the scale of the movement
    %% since this problem is the nub of the entire chapter.
    %% Need to justify the need for going to the trouble
    %% and define threshold for success (that couldn't be achieved!)

    The central requirement to accurately identify and quantitate the signal
    in the photobleached spot led us to pursue both cell biological approaches to
    minimise cell movement and and computational approaches to track imaged regions.

    \subsubsection{Contact inhibition at high cell density}

      Mammalian cells display contact inhibition,
      a cellular growth mechanism by which cells enter senescence and reduce motion
      when surrounded by other cells with no free space for movement.
      Although transformed cell lines lose this property,
      reduced space does place a restriction in the movement.
      %% Can you provide a semi-quantitative estimate of the cell density you mean?
      We attempted to use this effect to limit movement of cells
      by performing FRAP measurements with cells at higher confluence levels.
      We observed some decrease of movement for HeLa cells but not a complete immobilization.
      %% Can you provide a semi-quantitative estimate of the degree of reduction in movement?

      \begin{figure}
        %% We are only showing one cell rather than the whole field of
        %% of view because otherwise it's hard to notice the movement of
        %% individual cells. If we do display everything, we cell many
        %% nuclei that seem like their movement is smaller. If we do
        %% show it, we comment that we are unsure whether the movement
        %% is cellular or only nuclear.
        \centering
        \includegraphics[width=\textwidth]{kill-frap/results/confluent-hela.png}
        \captionIntro{Movement of confluent HeLa cells during FRAP experiment}
          {
            Cells reached confluence before the start of the
            experiment in an attempt to reduce motion. Instead, this caused
            cell nuclei to undergo heavy reshape as the cell apparently
            squeezes in between its neighbours. Half-nuclear FRAP performed in
            a confocal microscope over an interval of 8~hours. Top left panel
            is the pre-bleach image, while the others have a time-interval of
            21~minutes. Cells are a stable line derived from HeLa, expressing
            the H4~R45H mutant tagged with YFP.
          }
        \label{fig:kill-frap:confluent-hela}
      \end{figure}

      Tracking of the ROIs over time was still required \fref{fig:kill-frap:confluent-hela}.
      %% Isn't this out of order since CropReg section is below.
      %% Even if you did it in a different order, why not put the horse cells first.

    \subsubsection{Primary cell lines}

      Since HeLa cells have lost the ability to activate contact inhibition,
      we obtained an immortalised primary horse fibroblast cell line
      that displayed contact inhibition. Using this cell line we could
      maintain a layer of healthy cells covering a Petri dish for several days???
      after reaching confluence (data not shown).
      %% Cell growth was halted instead of becoming over-confluent ???what do you mean???
      %% Although obvious, it would be good to have some sort of image to justify the statement?

      However, primary cell line transfections have typically much lower efficiency rate and
      confluent cells have lower expression with each cell division.
      To balance transfection efficiency and expression we transfected cells at
      \SI{70}{\percent} confluence and imaged them after 3 days.

      Even after reaching confluence, we observed movement of transfected horse cells \fref{fig:kill-frap:confluent-horse}.
      %% Is there some sort of relative statement about the degree of reduction or residual movement?
      In addition, the observed movement was dramatically different in primary horse than in HeLa cells.
      Transfected horse cells displayed a helical motion about the vector of their movement
      whereas rotation of HeLa nuclei was mostly restricted to the $z$ axis.
      %% You need to define the coordinate system to talk about z axis!
      %% Do you mean vertical relative to the cell dish layer?

      \begin{figure}
        \centering
        \includegraphics[width=\textwidth]{kill-frap/results/confluent-horse.png}
        \captionIntro{Movement of confluent primary cells during FRAP experiment}
          {
            Primary horse fibrolasts display contact inhibition and halt growth
            once they reach confluence. However, this does not stop cell
            motion which can still be seen moving. In addition, when compared
            to the cancer cell line HeLa (\fref{fig:kill-frap:confluent-hela}),
            the horse fibroblasts frequently rotated around the $x$ and $y$
            axis. Circle FRAP was performed in a widefield microscope.
            Top left panel is the pre-bleach image, while the others have a
            time-interval of 15~minutes. Cells were transiently transfected
            and are expressing H2B type1-J tagged with EGFP.
          }
        \label{fig:kill-frap:confluent-horse}
      \end{figure}

    \subsubsection{Tracking of cell movement}

      As an alternative strategy, we implemented cell tracking
      in order to transform images into a common frame using
      consecutive image cropping and image registration.
      This approach was implemented as a program named CropReg.

      Nuclei of interest were tracked by template-based matching using normalized cross-correlation.
      Briefly, the nucleus to track was identified on the first frame and
      used as template against the image on the subsequent frame.
      For increased performance and robustness only the region surrounding the original position is used.
      %% Should you state approximately how large this region was?
      Sequentially applying this method created a stack of smaller images centred on the nuclei of interest.

      Achieving this functionality required implementing a ``coeff'' option
      for scaling in the \texttt{xcorr2} function in GNU Octave.
      This was contributed and released in version 1.2.0 of the Octave Forge signal package.
      %% TODO since there's more than one way to actually do the normalization,
      %% it might be a good idea to write down the actual math formula
      To correct for rotational movement around the $z$ axis,
      frames were aligned using rigid body geometric transformation using ImageJ plugin StackReg \citep{stackreg}.

      \begin{figure}
        \centering
        \includegraphics[width=\textwidth]{kill-frap/results/cropreg.png}
        %% imaging was done every 10 minutes, but we are skipping
        %% every other panel
        \captionIntro{Automatic tracking and alignment of moving cells}
          {
            Using CropReg, we successfully tracked individual cells during
            a time-series microscope experiment. The top left corner of each
            panel displays the tracked and aligned cell. Imaging was performed
            in a widefield microscope. Time interval between panels 20~minutes.
            Cells are a stable line derived from HeLa, expressing H3 tagged
            with YFP.
          }
        \label{fig:kill-frap:cropreg}
      \end{figure}

      Using this image processing approach we were able to track individual cell nuclei
      throughout an entire sequence of FRAP experiments \fref{fig:kill-frap:cropreg} provided
      that nuclei did not overlap.
      Although only a small minority of image sequences satisfied this requirement,
      it was possible to collect sufficient observations for FRAP calcultions.

  \subsection{Chromatin movement}

    While performing the FRAP experiments, we observed some movement
    within the cell nuclei. These could not be accounted for simple rotational
    movement around the $x$ or $y$ axis, and resembled more the movement
    of individual bodies within the nuclei.

    \subsubsection{Selection of \G1{} cells}
      %% There's no chemical equilibrium in S phase

      A possible cause of this chromatin movement comes from changes in
      the cell cycle phase. During the S~phase, the DNA is replicated,
      doubling the content of the chromatin.
      More importantly, this breaks
      a core assumption of FRAP, that the system remains in equilibrium
      during the entire experiment. This does not hold if the DNA, the
      binding sites for our model, duplicate in number.

      If the FRAP experiments can't be performed during S~phase and
      mitosis, we are limited to \G1{} and \G2{}. Considering
      the length of the HeLa cell cycle and the requirements to image
      for a time period of 8~hours, we are further limited to \G1{}.
      In addition, the FRAP experiment must be performed early in
      \G1{}~phase to avoid crossing over to the S~phase.

      %% The only reason this was required was because the LSM 510
      %% that we were using could not make Z stack and time lapse
      %% at the same time.
%      \begin{figure}
%        \centering
%        \missingfigure{Hela cells splitting}
%        \captionIntro{Picking cells at early G$_1$.}
%                     {We imaged cells that were entering mitosis and picked their
%                      daughter cells for the FRAP experiments. Because HeLa cells lift
%                      away from the dish during mitosis, opening the
%                      pinhole and set the Z-center in between the cell dividing plane
%                      and dish bottom was necessary. Ends up nothing being properly in focus but we
%                      can track things fine. Of course, some cells still floated away.}
%        %% TODO explicit parameters
%        \label{fig:kill-frap:picking-early-g1}
%      \end{figure}

      To do this, cells in mitosis were selected and tracked during 4~hours.
      After this time period, we used the daughter cells which we could be
      confident of being in early \G1{}.
      %% we also waited some 2 hours after mitosis since that's when cells
      %% unpack their chromosomes.

      During mitosis, HeLa cells form a sphere slightly above the plane of
      other cells, and keep a weak connection to the growth surface.
      Because of this, they easily detach, which is the basis for the
      mitotic shake-off method, and float away from the field of vision
      which requires a larger number
      of initial selected cells. In addition, to minimize any effect that
      may arise from imaging, it was done at minimal laser power and every
      30~minutes, just enough to allow manual tracking.
      Finally, since our system did not permit simultaneous Z-stack and time
      lapse imaging, and cells in mitosis are in a separate focal plane,
      imaging was performed with the pinhole sized to the max and focused
      in between the two planes. While this
      created very blurred images, it allowed to visualize all cells during
      the entire procedure.

      However, even after selecting cells in this cell cycle, movement within
      the bleach spot could still be observed.

    \subsubsection{Inverse FRAP}

      Due to the non-homogeneous nature of the chromatin, it was difficult
      to assess the total extent of the observed movement. To
      better visualize this, we performed inverse FRAP which allows us
      to track the movement of the bleach spot only.

      For this purpose, we replaced the EGFP tag in our H2B plasmid
      with photoactivatable GFP (PAGFP), a GFP derivative that requires
      activation by a specific wavelength to become fluorescent. This
      allows us to activate a specific spot of the nucleus and visualize
      its movement.

      Since PAGFP cannot be easily detected before photoactivation, cells
      were co-transfected with mCherry--\textalpha--tubulin which localises
      exclusively to the cytoplasm, giving an outline of the nuclear region
      \fref{fig:kill-frap:ifrap}.

      \begin{figure}
        \centering
        \subbottom[pre-activation]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/ifrap-pre.png}
          \label{fig:kill-frap:ifrap-pre}
        }
        \hfill
        \subbottom[post-activation]{
          \includegraphics[width=0.45\textwidth]
          {kill-frap/results/ifrap-post.png}
          \label{fig:kill-frap:ifrap-post}
        }
        \subbottom[activated spot over time]{
          \includegraphics[width=\textwidth]
          {kill-frap/results/ifrap.png}
          \label{fig:kill-frap:ifrap-timeframe}
        }
        \captionIntro{Inverse FRAP experiment showing chromatin movement}
          {
            HeLa cells co-transfected with mCherry--\textalpha--tubulin and
            H2B type1-J tagged with PAGFP.
            \subcaptionref{fig:kill-frap:ifrap-pre} The cell nucleus, target
            for photoactivation, can be easily identified as the ``empty''
            region via the mCherry channel on which would otherwise be an
            invisible feature on the GFP channel;
            \subcaptionref{fig:kill-frap:ifrap-post} spot after activation;
            \subcaptionref{fig:kill-frap:ifrap-timeframe} detail of the
            activated spot every 20~minutes. Rather than a gradual loss of
            fluorescence that maintains the circular shape, the activated spot
            kind of unfolds itself spreading the region of interest.
          }
        \label{fig:kill-frap:ifrap}
      \end{figure}

      Using this FRAP variant, the movement of chromatin was more noticeable.
      Rather than an homogeneous loss of fluorescence, the activated
      spot uncurled itself overtime with individual branches of
      localized PA-GFP appearing in the nuclei (\fref{fig:kill-frap:ifrap}).
